--- a/src/vpux_compiler/CMakeLists.txt
+++ b/src/vpux_compiler/CMakeLists.txt
@@ -156,7 +156,7 @@ endif()
 include(${PROJECT_SOURCE_DIR}/cmake/npu_compiler_version.cmake)
 npu_compiler_configure_version(${COMPILER_LIB})
 
-install_target_with_debug_files(${COMPILER_LIB} ${INSTALL_DESTINATION}/lib ${INSTALL_COMPONENT})
+install_target_with_debug_files(${COMPILER_LIB} lib ${INSTALL_COMPONENT})
 
 install(TARGETS ${COMPILER_LIB}
         ARCHIVE DESTINATION ${OV_CPACK_RUNTIMEDIR} COMPONENT ${VPUX_INTERNAL_COMPONENT}
--- a/src/vpux_driver_compiler/CMakeLists.txt
+++ b/src/vpux_driver_compiler/CMakeLists.txt
@@ -30,7 +30,7 @@ include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
 add_subdirectory(src)
 add_subdirectory(test)
 
-install(
+#[[install(
     FILES
         "${PROJECT_SOURCE_DIR}/src/vpux_driver_compiler/CHANGES.txt"
     DESTINATION ${INSTALL_DESTINATION}
@@ -40,4 +40,4 @@ install(
     FILES
         "${PROJECT_SOURCE_DIR}/src/vpux_driver_compiler/README.md"
     DESTINATION ${INSTALL_DESTINATION}
-    COMPONENT ${INSTALL_COMPONENT})
+    COMPONENT ${INSTALL_COMPONENT})]]
--- a/src/vpux_driver_compiler/cmake/build_manifest.cmake
+++ b/src/vpux_driver_compiler/cmake/build_manifest.cmake
@@ -48,5 +48,5 @@ configure_file(
 
 install(
     FILES ${CMAKE_CURRENT_BINARY_DIR}/build_manifest.json
-    DESTINATION ${INSTALL_DESTINATION}
+    DESTINATION share/intel-npu-compiler
     COMPONENT ${INSTALL_COMPONENT})
--- a/src/vpux_driver_compiler/src/vpux_compiler_l0/CMakeLists.txt
+++ b/src/vpux_driver_compiler/src/vpux_compiler_l0/CMakeLists.txt
@@ -63,13 +63,13 @@ add_dependencies(${TARGET_NAME} VPUXLoader)
 ov_add_api_validator_post_build_step(TARGET ${TARGET_NAME})
 
 # Install library related files to component
-install_target_with_debug_files(${TARGET_NAME} ${INSTALL_DESTINATION}/lib ${INSTALL_COMPONENT})
+install_target_with_debug_files(${TARGET_NAME} lib ${INSTALL_COMPONENT})
 
 # Install library header
 install(
         FILES
             "${PROJECT_SOURCE_DIR}/src/vpux_driver_compiler/include/npu_driver_compiler.h"
-        DESTINATION ${INSTALL_DESTINATION}
+        DESTINATION include
         COMPONENT ${INSTALL_COMPONENT})
 
 # Install the openvino_intel_npu_compiler library to the same path of mlir_compiler
--- a/src/vpux_driver_compiler/test/functional/CMakeLists.txt
+++ b/src/vpux_driver_compiler/test/functional/CMakeLists.txt
@@ -65,7 +65,7 @@ endif()
 if(ENABLE_PREBUILT_LLVM_MLIR_LIBS)
     find_package(MLIR REQUIRED CONFIG)
 endif()
-
+set_target_properties(${FUNCTIONAL_TARGET} PROPERTIES OUTPUT_NAME intel-npu-${FUNCTIONAL_TARGET})
 target_include_directories(${FUNCTIONAL_TARGET}
     SYSTEM PRIVATE
         $<BUILD_INTERFACE:${LLVM_SOURCE_DIR}/include>
@@ -77,5 +77,5 @@ ov_add_api_validator_post_build_step(TARGET vpuxCompilerL0Test)
 
 # Install functional test to CiD component
 if(BUILD_COMPILER_FOR_DRIVER)
-    install_target_with_debug_files(vpuxCompilerL0Test ${INSTALL_DESTINATION} ${INSTALL_COMPONENT})
+    install(TARGETS ${FUNCTIONAL_TARGET} RUNTIME DESTINATION bin COMPONENT ${INSTALL_COMPONENT})
 endif()
--- a/src/vpux_driver_compiler/test/smoke/CMakeLists.txt
+++ b/src/vpux_driver_compiler/test/smoke/CMakeLists.txt
@@ -12,6 +12,7 @@ include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
 # compilerTest as smoke test
 find_package(OpenVINO REQUIRED COMPONENTS Runtime)
 add_executable(compilerTest compilerTest.cpp)
+set_target_properties(compilerTest PROPERTIES OUTPUT_NAME intel-npu-compilerTest)
 target_link_libraries(compilerTest
     PRIVATE
         ${COMPILER_LOADER_DLL} # VCL function
@@ -23,6 +24,7 @@ ov_add_api_validator_post_build_step(TARGET compilerTest)
 
 # profilingTest to check profiling API
 add_executable(profilingTest profilingTest.c)
+set_target_properties(profilingTest PROPERTIES OUTPUT_NAME intel-npu-profilingTest)
 target_link_libraries(profilingTest
     PRIVATE
         ${COMPILER_LOADER_DLL})
@@ -30,6 +32,7 @@ ov_add_api_validator_post_build_step(TARGET profilingTest)
 
 # loaderTest to check npu_elf
 add_executable(loaderTest loaderTest.cpp)
+set_target_properties(loaderTest PROPERTIES OUTPUT_NAME intel-npu-loaderTest)
 target_link_libraries(loaderTest
     PRIVATE
         npu_elf
@@ -39,7 +42,7 @@ ov_add_api_validator_post_build_step(TARGET loaderTest)
 
 # Install functional test to CiD component
 if(BUILD_COMPILER_FOR_DRIVER)
-    install_target_with_debug_files(compilerTest ${INSTALL_DESTINATION} ${INSTALL_COMPONENT})
-    install_target_with_debug_files(profilingTest ${INSTALL_DESTINATION} ${INSTALL_COMPONENT})
-    install_target_with_debug_files(loaderTest ${INSTALL_DESTINATION} ${INSTALL_COMPONENT})
+    install_target_with_debug_files(compilerTest bin ${INSTALL_COMPONENT})
+    install_target_with_debug_files(profilingTest bin ${INSTALL_COMPONENT})
+    install_target_with_debug_files(loaderTest bin ${INSTALL_COMPONENT})
 endif()
