--- a/cmake/lib_install.cmake
+++ b/cmake/lib_install.cmake
@@ -79,7 +79,7 @@ ov_add_api_validator_post_build_step(TARGET ${TARGET_NAME})
 # loader folder | -> include/ | -> vpux_headers/ -> 1 hpp
 #               |             | -> vpux_loader/  -> 1 hpp
 
-install(DIRECTORY "${LOADER_DIR_INC}/vpux_headers"
+#[[install(DIRECTORY "${LOADER_DIR_INC}/vpux_headers"
         DESTINATION cid/vpux_elf/loader/include/
         COMPONENT ${CID_COMPONENT})
 
@@ -221,4 +221,4 @@ install(TARGETS npu_elf
         CONFIGURATIONS RelWithDebInfo
         LIBRARY DESTINATION cid/vpux_elf/lib/RelWithDebInfo COMPONENT ${CID_COMPONENT}
         ARCHIVE DESTINATION cid/vpux_elf/lib/RelWithDebInfo COMPONENT ${CID_COMPONENT}
-        RUNTIME DESTINATION cid/vpux_elf/lib/RelWithDebInfo COMPONENT ${CID_COMPONENT})
+        RUNTIME DESTINATION cid/vpux_elf/lib/RelWithDebInfo COMPONENT ${CID_COMPONENT})]]
