--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -235,11 +235,11 @@ if(JPEGLI_STATIC)
   # ourselves; for real use case we don't care about stdlib, as it is "granted",
   # so just linking all other libraries is fine.
   if (NOT MSVC AND NOT APPLE)
-    string(APPEND CMAKE_EXE_LINKER_FLAGS " -static")
+    #string(APPEND CMAKE_EXE_LINKER_FLAGS " -static")
   endif()
   if ((NOT WIN32 AND NOT APPLE) OR CYGWIN OR MINGW)
     set(CMAKE_FIND_LIBRARY_SUFFIXES .a)
-    string(APPEND CMAKE_EXE_LINKER_FLAGS " -static-libgcc -static-libstdc++")
+    #string(APPEND CMAKE_EXE_LINKER_FLAGS " -static-libgcc -static-libstdc++")
   endif()
 endif()  # JPEGLI_STATIC
 
