diff --git a/UI/Qt/CMakeLists.txt b/UI/Qt/CMakeLists.txt
index 93fa801ebf9..8474b25d770 100644
--- a/UI/Qt/CMakeLists.txt
+++ b/UI/Qt/CMakeLists.txt
@@ -15,5 +15,5 @@ target_sources(ladybird PRIVATE
     ladybird.qrc
     main.cpp
 )
-target_link_libraries(ladybird PRIVATE Qt::Core Qt::Gui Qt::Widgets)
+target_link_libraries(ladybird PRIVATE Qt::Core Qt::Gui Qt::Widgets -Wl,--whole-archive harfbuzz fontconfig -Wl,--no-whole-archive)
 create_ladybird_bundle(ladybird)
