diff -ruN albert.orig/plugins/python/CMakeLists.txt albert/plugins/python/CMakeLists.txt
--- albert.orig/plugins/python/CMakeLists.txt	2024-08-22 02:00:21.242253759 +0300
+++ albert/plugins/python/CMakeLists.txt	2024-08-22 02:00:39.342677409 +0300
@@ -17,7 +17,6 @@
 install(
     DIRECTORY "plugins/"
     DESTINATION "${CMAKE_INSTALL_DATADIR}/albert/${PROJECT_NAME}/plugins"
-    REGEX "/\\..*" EXCLUDE  # exclude hidden files
     PATTERN "__pycache__" EXCLUDE
 )
 
