--- old/install.js	2024-07-21 11:42:13.920979705 +0900
+++ new/install.js	2024-07-21 11:42:36.466023502 +0900
@@ -43,9 +43,5 @@ DEST_DIR = argv.DEST_DIR;
 export async function install() {
   console.log(`Copying nody-greeter to "${DEST_DIR}"...`);
   await makeCopy(INSTALL_ROOT, DEST_DIR);
-  fs.createSymlinkSync(
-    path.join(DEST_DIR, "opt/nody-greeter/nody-greeter"),
-    path.join(DEST_DIR, PREFIX, "bin/nody-greeter")
-  );
   console.log("\x1b[92mSUCCESS!!\x1b[0m");
 }
