--- a/src/optBase.h	2025-05-02 09:43:20.000000000 +0200
+++ b/src/optBase.h	2025-10-10 01:48:53.008709281 +0200
@@ -150,8 +150,7 @@
     { PM_MTID_DSMATTE,       "DBSMATT",  N_("Double-Sided Matte Paper")},
     { PM_MTID_CLPHOTO,       "CLPP",     N_("ColorLife Photo Paper")},
     { PM_MTID_ECOPHOTO,      "EPHOTO",   N_("Economy Photo Paper")},
-//    { PM_MTID_VELVETFINEART, "VELVET_FINE", N_("Velvet Fine Art Paper")},
-    { PM_MTID_MATTE, 	     "VELVET_FINE", N_("Velvet Fine Art Paper")},	/* Treated as "EPSON Matte" */
+    { PM_MTID_VELVETFINEART, "VELVET_FINE", N_("Velvet Fine Art Paper")},
     { PM_MTID_PROOFSEMI,     "PROOF_S",  N_("Proofing Paper Semimatte")},
     { PM_MTID_HAGAKIRECL,    "RCPC",     N_("Recycled Paper Postcard")},
     { PM_MTID_HAGAKIINKJET,  "IJPC",     N_("Ink Jet Postcard")},
