diff --git a/util/grub-install.c b/util/grub-install.c
index 53b464804..2bf02a3b6 100644
--- a/util/grub-install.c
+++ b/util/grub-install.c
@@ -446,6 +446,8 @@ probe_mods (grub_disk_t disk)
     {
       grub_util_cryptodisk_get_abstraction (disk,
 					    push_cryptodisk_module, NULL);
+      /* HACK: always push argon2 */
+      grub_install_push_module ("argon2");
       have_abstractions = 1;
       have_cryptodisk = 1;
     }
