diff --git a/setup.py b/setup.py
index 909a3ff..6c20685 100644
--- a/setup.py
+++ b/setup.py
@@ -1,16 +1,12 @@
-from codecs     import open
-from inspect    import getsource
 from os.path    import abspath, dirname, join
 from setuptools import setup
 
-here = abspath(dirname(getsource(lambda:0)))
-
-with open(join(here, 'README.rst'), encoding='utf-8') as f:
+with open('README.rst', encoding='utf-8') as f:
     long_description = f.read()
 
 setup(name             = 'playsound',
       version          = '1.3.0',
-      description      = long_description.splitlines()[2][1:-1],
+      description      = 'Pure Python, cross platform, single function module with no dependencies for playing sounds',
       long_description = long_description,
       url              = 'https://github.com/TaylorSMarks/playsound',
       author           = 'Taylor Marks',
