diff --git a/src/exm-detail-view.c b/src/exm-detail-view.c
index 887fbf3..10a458e 100644
--- a/src/exm-detail-view.c
+++ b/src/exm-detail-view.c
@@ -25,6 +25,8 @@
 #include "exm-info-bar.h"
 #include "exm-comment-tile.h"
 #include "exm-comment-dialog.h"
+#include "exm-install-button.h"
+#include "exm-screenshot.h"

 #include "web/exm-data-provider.h"
 #include "web/exm-image-resolver.h"
@@ -682,6 +684,10 @@ exm_detail_view_init (ExmDetailView *self)
 {
     GtkAdjustment *adj;

+    g_type_ensure (EXM_TYPE_INSTALL_BUTTON);
+    g_type_ensure (EXM_TYPE_SCREENSHOT);
+    g_type_ensure (EXM_TYPE_INFO_BAR);
+
     gtk_widget_init_template (GTK_WIDGET (self));

     self->provider = exm_data_provider_new ();
