diff --git a/src/exm-extension-row.c b/src/exm-extension-row.c
index 63c0fa2..c769e68 100644
--- a/src/exm-extension-row.c
+++ b/src/exm-extension-row.c
@@ -234,7 +234,7 @@ on_state_changed (GtkSwitch        *toggle,
                   gboolean          state,
                   ExmExtensionRow  *self)
 {
-    g_return_if_fail (EXM_IS_EXTENSION_ROW (self));
+    g_return_val_if_fail (EXM_IS_EXTENSION_ROW (self), FALSE);

     g_assert (self->ext_toggle == toggle);
