--- hal-emsec/plugins/graph_algorithm/src/algorithms/shortest_path.cpp.orig	2025-12-04 21:12:40.113050627 -0300
+++ hal-emsec/plugins/graph_algorithm/src/algorithms/shortest_path.cpp	2025-12-04 21:13:20.891525976 -0300
@@ -122,7 +122,7 @@
                 return ERR(igraph_strerror(res));
             }
 
-            if (auto res = igraph_get_shortest_paths(graph->get_graph(), &paths_raw, nullptr, from_vertex, v_sel, mode, nullptr, nullptr); res != IGRAPH_SUCCESS)
+            if (auto res = igraph_get_shortest_paths(graph->get_graph(), nullptr, &paths_raw, nullptr, from_vertex, v_sel, mode, nullptr, nullptr); res != IGRAPH_SUCCESS)
             {
                 igraph_vs_destroy(&v_sel);
                 igraph_vector_int_list_destroy(&paths_raw);
@@ -264,7 +264,7 @@
                 return ERR(igraph_strerror(res));
             }
 
-            if (auto res = igraph_get_all_shortest_paths(graph->get_graph(), &paths_raw, nullptr, nullptr, from_vertex, v_sel, mode); res != IGRAPH_SUCCESS)
+            if (auto res = igraph_get_all_shortest_paths(graph->get_graph(), nullptr, &paths_raw, nullptr, nullptr, from_vertex, v_sel, mode); res != IGRAPH_SUCCESS)
             {
                 igraph_vs_destroy(&v_sel);
                 igraph_vector_int_list_destroy(&paths_raw);
@@ -292,4 +292,4 @@
             return OK(paths);
         }
     }    // namespace graph_algorithm
-}    // namespace hal
\ No newline at end of file
+}    // namespace hal
--- hal-emsec/plugins/hawkeye/src/candidate_search.cpp.orig	2025-12-04 21:38:31.363694493 -0300
+++ hal-emsec/plugins/hawkeye/src/candidate_search.cpp	2025-12-04 21:42:36.890911465 -0300
@@ -113,7 +113,7 @@
 
                     if (previous_size < current_size)
                     {
-                        IGRAPH_CHECK(igraph_vector_int_swap(previous_hood_p, current_hood_p));
+                        igraph_vector_int_swap(previous_hood_p, current_hood_p);
                         igraph_vector_int_clear(current_hood_p);
 
                         memset(added, false, no_of_nodes * sizeof(igraph_bool_t));
@@ -122,7 +122,7 @@
                         {
                             igraph_integer_t actnode = VECTOR(*previous_hood_p)[j];
                             igraph_vector_int_clear(&tmp);
-                            IGRAPH_CHECK(igraph_neighbors(graph, &tmp, actnode, IGRAPH_OUT));
+                            igraph_neighbors(graph, &tmp, actnode, IGRAPH_OUT, IGRAPH_LOOPS_ONCE, static_cast<igraph_bool_t>(1));
 
                             for (k = 0; k < igraph_vector_int_size(&tmp); k++)
                             {
@@ -208,8 +208,8 @@
                     if (previous_size < current_size || current_size == 1)
                     {
                         // move current objects to previous
-                        IGRAPH_CHECK(igraph_vector_int_swap(previous_hood_p, current_hood_p));
-                        IGRAPH_CHECK(igraph_vector_int_swap(previous_component_p, current_component_p));
+                        igraph_vector_int_swap(previous_hood_p, current_hood_p);
+                        igraph_vector_int_swap(previous_component_p, current_component_p);
                         igraph_vector_int_clear(current_hood_p);
                         igraph_vector_int_clear(current_component_p);
 
@@ -221,7 +221,7 @@
                         {
                             igraph_integer_t actnode = VECTOR(*previous_hood_p)[j];
                             igraph_vector_int_clear(&tmp);
-                            IGRAPH_CHECK(igraph_neighbors(graph, &tmp, actnode, IGRAPH_OUT));
+                            igraph_neighbors(graph, &tmp, actnode, IGRAPH_OUT, IGRAPH_LOOPS_ONCE, static_cast<igraph_bool_t>(1));
 
                             for (k = 0; k < igraph_vector_int_size(&tmp); k++)
                             {
@@ -669,4 +669,4 @@
     std::map<hawkeye::DetectionConfiguration::Components, std::string> EnumStrings<hawkeye::DetectionConfiguration::Components>::data = {
         {hawkeye::DetectionConfiguration::Components::NONE, "NONE"},
         {hawkeye::DetectionConfiguration::Components::CHECK_SCC, "CHECK_SCC"}};
-}    // namespace hal
\ No newline at end of file
+}    // namespace hal
