diff --git i/lib/astal/gtk4/src/meson.build w/lib/astal/gtk4/src/meson.build
index 3bd6a2f..ca6afa8 100644
--- i/lib/astal/gtk4/src/meson.build
+++ w/lib/astal/gtk4/src/meson.build
@@ -40,9 +40,10 @@ lib = library(
   vala_args: ['--vapi-comments'],
   vala_header: meson.project_name() + '.h',
   vala_vapi: meson.project_name() + '-' + api_version + '.vapi',
+  vala_gir: gir,
   version: meson.project_version(),
   install: true,
-  install_dir: [true, true, true],
+  install_dir: [true, true, true, true],
 )
 
 pkgs = []
@@ -50,22 +51,22 @@ foreach dep : deps
   pkgs += ['--pkg=' + dep.name()]
 endforeach
 
-gir_tgt = custom_target(
-  gir,
-  command: [
-    find_program('python3'),
-    girpy,
-    meson.project_name(),
-    gir + ':src/' + gir,
-  ]
-  + pkgs
-  + sources,
-  input: sources,
-  depends: lib,
-  output: gir,
-  install: true,
-  install_dir: get_option('datadir') / 'gir-1.0',
-)
+# gir_tgt = custom_target(
+#   gir,
+#   command: [
+#     find_program('python3'),
+#     girpy,
+#     meson.project_name(),
+#     gir + ':src/' + gir,
+#   ]
+#   + pkgs
+#   + sources,
+#   input: sources,
+#   depends: lib,
+#   output: gir,
+#   install: true,
+#   install_dir: get_option('datadir') / 'gir-1.0',
+# )
 
 custom_target(
   typelib,
@@ -77,7 +78,7 @@ custom_target(
   ],
   input: lib,
   output: typelib,
-  depends: [lib, gir_tgt],
+  depends: [lib],
   install: true,
   install_dir: libdir / 'girepository-1.0',
 )
