diff --git a/src/Views/NoteContentView.vala b/src/Views/NoteContentView.vala
index ba6c608..702faae 100644
--- a/src/Views/NoteContentView.vala
+++ b/src/Views/NoteContentView.vala
@@ -247,7 +247,7 @@ public class Notejot.NoteContentView : View {
             if (((Adw.Leaflet)MiscUtils.find_ancestor_of_type<Adw.Leaflet>(this)).folded) {
                 back2_button.visible = false;
             } else {
-                back2_button.visible = ((MainWindow)MiscUtils.find_ancestor_of_type<MainWindow>(this)).sgrid.get_visible_child_name () == "notegrid" != false ? true : false;
+                back2_button.visible = ((MainWindow)MiscUtils.find_ancestor_of_type<MainWindow>(this)).sgrid.get_visible_child_name () == "notegrid";
             }
             back2_button.clicked.connect (() => {
                 ((Adw.Leaflet)MiscUtils.find_ancestor_of_type<Adw.Leaflet>(this)).set_visible_child (((MainWindow)MiscUtils.find_ancestor_of_type<MainWindow>(this)).sgrid);
-- 
2.41.0
