--- a/src/frontends/onnx/frontend/src/frontend.cpp
+++ b/src/frontends/onnx/frontend/src/frontend.cpp
@@ -14,7 +14,7 @@
 #ifdef OV_PROTOBUF_ABSL_IS_USED
 #    include <absl/log/globals.h>
 #else
-#    include <google/protobuf/stubs/logging.h>
+//#    include <google/protobuf/stubs/logging.h>
 #endif
 
 #include <fstream>
@@ -55,7 +55,7 @@ ONNX_FRONTEND_C_API void* get_front_end_data() {
 #    ifdef OV_PROTOBUF_ABSL_IS_USED
     absl::SetGlobalVLogLevel(0);
 #    else
-    google::protobuf::SetLogHandler(nullptr);
+    //google::protobuf::SetLogHandler(nullptr);
 #    endif
 #endif
     return res;
--- a/src/frontends/paddle/src/frontend.cpp
+++ b/src/frontends/paddle/src/frontend.cpp
@@ -16,7 +16,7 @@
 #ifdef OV_PROTOBUF_ABSL_IS_USED
 #    include <absl/log/globals.h>
 #else
-#    include <google/protobuf/stubs/logging.h>
+//#    include <google/protobuf/stubs/logging.h>
 #endif
 
 #include <fstream>
@@ -602,7 +602,7 @@ PADDLE_C_API void* get_front_end_data() {
 #    ifdef OV_PROTOBUF_ABSL_IS_USED
     absl::SetGlobalVLogLevel(0);
 #    else
-    google::protobuf::SetLogHandler(nullptr);
+    //google::protobuf::SetLogHandler(nullptr);
 #    endif
 #endif
     return res;
