--- x2goserver-4.1.0.3/x2goserver-common/Makefile	2020-05-29 13:47:22.000000000 +0200
+++ x2goserver-4.1.0.3/x2goserver-common/Makefile.new	2020-05-29 13:49:19.785916164 +0200
@@ -70,7 +70,7 @@
 install_man:
 	$(INSTALL_DIR) $(DESTDIR)$(MANDIR)/man5
 	$(INSTALL_FILE) man/man5/*.5           $(DESTDIR)$(MANDIR)/man5
-	gzip -f $(DESTDIR)$(MANDIR)/man5/x2go*.5
+	gzip -f -n $(DESTDIR)$(MANDIR)/man5/x2go*.5
 
 install_version:
 	$(INSTALL_DIR) $(DESTDIR)$(SHAREDIR)
--- x2goserver-4.1.0.3/x2goserver/Makefile	2020-05-29 13:47:22.000000000 +0200
+++ x2goserver-4.1.0.3/x2goserver/Makefile.new	2020-05-29 13:50:51.578356020 +0200
@@ -80,10 +80,10 @@
 	$(INSTALL_DIR) $(DESTDIR)$(MANDIR)
 	$(INSTALL_DIR) $(DESTDIR)$(MANDIR)/man5
 	$(INSTALL_FILE) man/man5/*.5           $(DESTDIR)$(MANDIR)/man5
-	gzip -f $(DESTDIR)$(MANDIR)/man5/x2go*.5
+	gzip -f -n $(DESTDIR)$(MANDIR)/man5/x2go*.5
 	$(INSTALL_DIR) $(DESTDIR)$(MANDIR)/man8
 	$(INSTALL_FILE) man/man8/*.8           $(DESTDIR)$(MANDIR)/man8
-	gzip -f $(DESTDIR)$(MANDIR)/man8/x2go*.8
+	gzip -f -n $(DESTDIR)$(MANDIR)/man8/x2go*.8
 
 install_version:
 	$(INSTALL_DIR) $(DESTDIR)$(SHAREDIR)
--- x2goserver-4.1.0.3/x2goserver-x2goagent/Makefile	2020-05-29 13:47:22.000000000 +0200
+++ x2goserver-4.1.0.3/x2goserver-x2goagent/Makefile.new	2020-05-29 13:52:31.241005436 +0200
@@ -75,7 +75,7 @@
 	$(INSTALL_DIR) $(DESTDIR)$(MANDIR)
 	$(INSTALL_DIR) $(DESTDIR)$(MANDIR)/man1
 	$(INSTALL_FILE) man/man1/*.1           $(DESTDIR)$(MANDIR)/man1
-	gzip -f $(DESTDIR)$(MANDIR)/man1/x2go*.1
+	gzip -f -n $(DESTDIR)$(MANDIR)/man1/x2go*.1
 
 install_version:
 	$(INSTALL_DIR) $(DESTDIR)$(SHAREDIR)
--- x2goserver-4.1.0.3/x2goserver-printing/Makefile	2020-05-29 13:47:22.000000000 +0200
+++ x2goserver-4.1.0.3/x2goserver-printing/Makefile.new	2020-05-29 13:58:14.460132349 +0200
@@ -62,7 +62,7 @@
 	$(INSTALL_DIR) $(DESTDIR)$(MANDIR)
 	$(INSTALL_DIR) $(DESTDIR)$(MANDIR)/man8
 	$(INSTALL_FILE) man/man8/*.8           $(DESTDIR)$(MANDIR)/man8
-	gzip -f $(DESTDIR)$(MANDIR)/man8/x2go*.8
+	gzip -f -n $(DESTDIR)$(MANDIR)/man8/x2go*.8
 
 install_version:
 	$(INSTALL_DIR) $(DESTDIR)$(SHAREDIR)
--- x2goserver-4.1.0.3/x2goserver-extensions/Makefile	2020-05-29 13:47:22.000000000 +0200
+++ x2goserver-4.1.0.3/x2goserver-extensions/Makefile.new	2020-05-29 13:59:21.731921736 +0200
@@ -94,7 +94,7 @@
 	$(INSTALL_DIR) $(DESTDIR)$(MANDIR)
 	$(INSTALL_DIR) $(DESTDIR)$(MANDIR)/man8
 	$(INSTALL_FILE) man/man8/*.8           $(DESTDIR)$(MANDIR)/man8
-	gzip -f $(DESTDIR)$(MANDIR)/man8/x2go*.8
+	gzip -f -n $(DESTDIR)$(MANDIR)/man8/x2go*.8
 
 install_version:
 	$(INSTALL_DIR) $(DESTDIR)$(SHAREDIR)
--- x2goserver-4.1.0.3/x2goserver-fmbindings/Makefile	2020-05-29 13:47:22.000000000 +0200
+++ x2goserver-4.1.0.3/x2goserver-fmbindings/Makefile.new	2020-05-29 14:02:45.167334027 +0200
@@ -66,7 +66,7 @@
 	$(INSTALL_DIR) $(DESTDIR)$(MANDIR)
 	$(INSTALL_DIR) $(DESTDIR)$(MANDIR)/man8
 	$(INSTALL_FILE) man/man8/*.8           $(DESTDIR)$(MANDIR)/man8
-	gzip -f $(DESTDIR)$(MANDIR)/man8/x2go*.8
+	gzip -f -n $(DESTDIR)$(MANDIR)/man8/x2go*.8
 
 install_version:
 	$(INSTALL_DIR) $(DESTDIR)$(SHAREDIR)
--- x2goserver-4.1.0.3/x2goserver-desktopsharing/Makefile	2020-05-29 13:47:22.000000000 +0200
+++ x2goserver-4.1.0.3/x2goserver-desktopsharing/Makefile.new	2020-05-29 14:03:36.378699088 +0200
@@ -64,7 +64,7 @@
 	$(INSTALL_DIR) $(DESTDIR)$(MANDIR)
 	$(INSTALL_DIR) $(DESTDIR)$(MANDIR)/man8
 	$(INSTALL_FILE) man/man8/*.8           $(DESTDIR)$(MANDIR)/man8
-	gzip -f $(DESTDIR)$(MANDIR)/man8/x2go*.8
+	gzip -f -n $(DESTDIR)$(MANDIR)/man8/x2go*.8
 
 install_version:
 	$(INSTALL_DIR) $(DESTDIR)$(SHAREDIR)
