--- a/serial-stubs/tools/list_ports.pyi	2026-06-01 20:00:49.144975412 +0200
+++ b/serial-stubs/tools/list_ports.pyi	2026-06-01 20:01:53.968974086 +0200
@@ -6,8 +6,13 @@
 
 if sys.platform == "win32":
     from serial.tools.list_ports_windows import comports as comports
+    from serial.tools.list_ports_common import ListPortInfo
 else:
     from serial.tools.list_ports_posix import comports as comports
+    from serial.tools.list_ports_linux import SysFS

-def grep(regexp: str | re.Pattern[str], include_links: bool = False) -> Generator[ListPortInfo]: ...
+if sys.platform == "win32":
+    def grep(regexp: str | re.Pattern[str], include_links: bool = False) -> Generator[ListPortInfo]: ...
+else:
+    def grep(regexp: str | re.Pattern[str], include_links: bool = False) -> Generator[SysFS]: ...
 def main() -> None: ...
 