From ff5b8a5c11af53a087d949169dc73597bfcc7f26 Mon Sep 17 00:00:00 2001
From: Stelios Tsampas <loathingkernel@gmail.com>
Date: Thu, 6 Jun 2024 21:47:33 +0300
Subject: [PATCH] AUR: run make_requests before configure

---
 make/rules-wine-requests.mk | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/make/rules-wine-requests.mk b/make/rules-wine-requests.mk
index 4244fdac..28a0848d 100644
--- a/make/rules-wine-requests.mk
+++ b/make/rules-wine-requests.mk
@@ -23,8 +23,8 @@ $$(OBJ)/.$(1)-wine-requests: | $$(OBJ)/.$(1)-post-source
 	cd "$$($(2)_SRC)" && tools/make_requests
 	touch $$@
 
-$$(OBJ)/.$(1)-build32: $$(OBJ)/.$(1)-wine-requests
-$$(OBJ)/.$(1)-build64: $$(OBJ)/.$(1)-wine-requests
+$$(OBJ)/.$(1)-configure32: $$(OBJ)/.$(1)-wine-requests
+$$(OBJ)/.$(1)-configure64: $$(OBJ)/.$(1)-wine-requests
 endef
 
 rules-wine-requests = $(call create-rules-wine-requests,$(1),$(call toupper,$(1)))
-- 
2.45.2

