From 345cd77f55b6dd3ac0570ffe54df6d2a1c4873d6 Mon Sep 17 00:00:00 2001
From: Stelios Tsampas <loathingkernel@gmail.com>
Date: Sat, 11 May 2024 21:58:52 +0300
Subject: [PATCH 7/8] AUR: Disable incompatible-pointer-types in Wine (GCC 14)

---
 Makefile.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Makefile.in b/Makefile.in
index 3f1c9ff5..b3f0784e 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -626,6 +626,9 @@ WINE_CONFIGURE_ARGS64 = --enable-win64 \
   VKD3D_PE_CFLAGS="-I$(VULKAN_HEADERS_DST64)/include -I$(VKD3D_DST64)/include/vkd3d" \
   VKD3D_PE_LIBS="-L$(VKD3D_DST64)/bin -l:libvkd3d-1.dll -l:libvkd3d-shader-1.dll"
 
+WINE_CFLAGS = \
+  -Wno-incompatible-pointer-types \
+
 WINE_DEPENDS = gst_orc gstreamer gst_base vkd3d
 
 $(eval $(call rules-source,wine,$(SRCDIR)/wine))
-- 
2.48.1

