From 7575631e65c8b49fdb7a3665ade665dcea643fc1 Mon Sep 17 00:00:00 2001
From: Stelios Tsampas <loathingkernel@gmail.com>
Date: Thu, 6 Jun 2024 19:47:02 +0300
Subject: [PATCH 8/8] AUR: Fix calling make-requests where it shouldn't be
 called

---
 Makefile.in | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index 63e19456..3f5b912f 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -639,7 +639,6 @@ $(eval $(call rules-wine-requests,wine))
 $(OBJ)/.wine-post-source:
 	cd $(WINE_SRC) && dlls/winevulkan/make_vulkan -x vk.xml
 	-cd $(WINE_SRC) && tools/make_specfiles
-	-cd $(WINE_SRC) && tools/make_requests
 	touch $@
 
 $(OBJ)/.wine-post-build64:
-- 
2.46.1

