diff --git a/meson.build b/meson.build
index ce2961ae..9305da5d 100644
--- a/meson.build
+++ b/meson.build
@@ -17,7 +17,7 @@
 project(
   'vte',
   ['c', 'cpp'],
-  version: '0.76.4',
+  version: '10000.76.4',
   license: ['LGPL-3.0-or-later', 'GPL-3.0-or-later'],
   default_options: [
     'buildtype=release',
@@ -62,7 +62,7 @@ liblz4_req_version        = '1.9'
 
 # API
 
-vte_api_major_version = 2
+vte_api_major_version = 10002
 vte_api_minor_version = 91
 vte_api_version = '@0@.@1@'.format(vte_api_major_version, vte_api_minor_version)
 vte_api_name = 'vte-@0@.@1@'.format(vte_api_major_version, vte_api_minor_version)
diff --git a/src/meson.build b/src/meson.build
index 3f89f492..9a0ff506 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -726,7 +726,7 @@ vte_csh = configure_file(
 
 vte_sh = configure_file(
   input: 'vte.sh.in',
-  output: '@BASENAME@',
+  output: '100007604-@BASENAME@',
   configuration: {
     'libexecdir': vte_prefix / vte_libexecdir,
   },
diff --git a/src/vte.sh.in b/src/vte.sh.in
index 2328a9ec..f4210af4 100644
--- a/src/vte.sh.in
+++ b/src/vte.sh.in
@@ -20,7 +20,7 @@
 [[ $- == *i* ]] || return 0
 
 # Not running under vte?
-[ "${VTE_VERSION:-0}" -ge 3405 ] || return 0
+[ "${VTE_VERSION:-0}" -ge 100007604 ] || return 0
 
 # TERM not supported?
 case "$TERM" in
