--- a/app/build.gradle.kts	2025-04-28 09:19:01.183117452 +0200
+++ b/app/build.gradle.kts	2025-04-28 09:17:50.418146904 +0200
@@ -414,12 +414,12 @@
 }
 tasks.register("includeProcessingResources"){
     dependsOn(
-        "includeJdk",
+        //"includeJdk",
         "includeCore",
         "includeJavaMode",
         "includeSharedAssets",
-        "includeProcessingExamples",
-        "includeProcessingWebsiteExamples",
+        //"includeProcessingExamples",
+        //"includeProcessingWebsiteExamples",
         "includeJavaModeResources",
         "renameWindres"
     )
