--- a/app/build.gradle.kts	2025-04-25 14:17:52.284598476 +0200
+++ b/app/build.gradle.kts	2025-04-25 14:17:23.982869026 +0200
@@ -425,10 +425,10 @@
     dependsOn(
         "includeCore",
         "includeJavaMode",
-        "includeJdk",
+        //"includeJdk",
         "includeSharedAssets",
-        "includeProcessingExamples",
-        "includeProcessingWebsiteExamples",
+        //"includeProcessingExamples",
+        //"includeProcessingWebsiteExamples",
         "includeJavaModeResources",
         "renameWindres"
     )
@@ -511,10 +511,10 @@
         dependsOn(
             "includeCore",
             "includeJavaMode",
-            "includeJdk",
+            //"includeJdk",
             "includeSharedAssets",
-            "includeProcessingExamples",
-            "includeProcessingWebsiteExamples",
+            //"includeProcessingExamples",
+            //"includeProcessingWebsiteExamples",
             "includeJavaModeResources",
             "renameWindres"
         )
