diff --git i/src/Modules/Main.lua w/src/Modules/Main.lua
index 87dd2744c..819322497 100644
--- i/src/Modules/Main.lua
+++ w/src/Modules/Main.lua
@@ -328,7 +328,7 @@ end
 
 function main:OnFrame()
 	self.screenW, self.screenH = GetScreenSize()
-	self.screenScale = GetScreenScale()
+	self.screenScale = 1.0
 	if self.screenScale ~= 1.0 then
 		self.screenW = math.floor(self.screenW / self.screenScale)
 		self.screenH = math.floor(self.screenH / self.screenScale)
