diff --git a/src/Launch.lua b/src/Launch.lua
index 0775ec4dc..a5113655c 100644
--- a/src/Launch.lua
+++ b/src/Launch.lua
@@ -56,7 +56,7 @@ function launch:OnInit()
 	if localManXML and not self.versionBranch and not self.versionPlatform then
 		-- Looks like a remote manifest, so we're probably running from a repository
 		-- Enable dev mode to disable updates and set user path to be the script path
-		self.devMode = true
+		self.devMode = false
 	end
 	local installedFile = io.open("installed.cfg", "r")
 	if installedFile then
@@ -324,6 +324,8 @@ function launch:ApplyUpdate(mode)
 end
 
 function launch:CheckForUpdate(inBackground)
+	return
+	--[[
 	if self.updateCheckRunning then
 		return
 	end
@@ -340,6 +342,7 @@ function launch:CheckForUpdate(inBackground)
 		self.updateCheckRunning = true
 	end
 	update:close()
+	--]]
 end
 
 function launch:ShowPrompt(r, g, b, str, func)
diff --git a/src/Modules/Main.lua b/src/Modules/Main.lua
index 39385efab..144e072ef 100644
--- a/src/Modules/Main.lua
+++ b/src/Modules/Main.lua
@@ -73,7 +73,7 @@ function main:Init()
 	if not ignoreBuild then
 		self:SetMode("BUILD", false, "Unnamed build")
 	end
-	if launch.devMode or (GetScriptPath() == GetRuntimePath() and not launch.installedMode) then
+	if launch.devMode and (GetScriptPath() == GetRuntimePath() and not launch.installedMode) then
 		-- If running in dev mode or standalone mode, put user data in the script path
 		self.userPath = GetScriptPath().."/"
 	else
