--- a/source/CMakeLists.txt
+++ b/source/CMakeLists.txt
@@ -104,7 +104,7 @@ MESSAGE(STATUS "Target CPU\t\t\t: ${UAVS3D_TARGET_CPU}")
 find_package(Threads REQUIRED)
 set(prefix "${CMAKE_INSTALL_PREFIX}")
 set(includedir "include")
-set(libdir "lib")
+set(libdir "lib/uavs3d-10bit")
 set(pc_file "${CONFIG_DIR}/${LIBNAME}.pc")
 
 set(CMAKE_INSTALL_INCLUDE_DIR "${CMAKE_INSTALL_PREFIX}/${includedir}")
