--- a/shared/generate_cpp_array/source/generate_cpp_array.cpp
+++ b/shared/generate_cpp_array/source/generate_cpp_array.cpp
11:51:21.345537972 -0300
@@ -5,6 +5,7 @@
  *
  */
 
+#include <cstdint>
 #include <fstream>
 #include <iomanip>
 #include <iostream>
--- a/shared/offline_compiler/source/decoder/iga_wrapper.h
+++ b/shared/offline_compiler/source/decoder/iga_wrapper.h
@@ -9,6 +9,7 @@
 
 #include "igfxfmid.h"
 
+#include <cstdint>
 #include <memory>
 #include <string>
 
--- a/shared/offline_compiler/source/ocloc_arg_helper.h
+++ b/shared/offline_compiler/source/ocloc_arg_helper.h
@@ -11,6 +11,7 @@
 #include "shared/source/utilities/const_stringref.h"
 
 #include <algorithm>
+#include <cstdint>
 #include <fstream>
 #include <map>
 #include <memory>
--- a/shared/source/debugger/debugger.h
+++ b/shared/source/debugger/debugger.h
@@ -6,6 +6,7 @@
  */
 
 #pragma once
+#include <cstdint>
 #include <memory>
 namespace NEO {
 struct HardwareInfo;
--- a/shared/source/gmm_helper/gmm_helper.h
+++ b/shared/source/gmm_helper/gmm_helper.h
@@ -6,6 +6,7 @@
  */
 
 #pragma once
+#include <cstdint>
 #include <memory>
 
 namespace NEO {
--- a/shared/source/os_interface/device_factory.h
+++ b/shared/source/os_interface/device_factory.h
@@ -6,6 +6,7 @@
  */
 
 #pragma once
+#include <cstdint>
 #include <memory>
 #include <string>
 #include <vector>
--- a/shared/source/os_interface/os_memory.h
+++ b/shared/source/os_interface/os_memory.h
@@ -6,6 +6,7 @@
  */
 
 #pragma once
+#include <cstdint>
 #include <memory>
 #include <vector>
 
--- a/shared/source/os_interface/os_time.h
+++ b/shared/source/os_interface/os_time.h
@@ -6,6 +6,7 @@
  */
 
 #pragma once
+#include <cstdint>
 #include <memory>
 #include <optional>
 
--- a/shared/source/program/program_info.h
+++ b/shared/source/program/program_info.h
@@ -8,6 +8,7 @@
 #pragma once
 #include "shared/source/utilities/arrayref.h"
 
+#include <cstdint>
 #include <cstddef>
 #include <memory>
 #include <string>
--- a/shared/source/utilities/software_tags.h
+++ b/shared/source/utilities/software_tags.h
@@ -8,6 +8,7 @@
 #pragma once
 #include "shared/source/helpers/string.h"
 
+#include <cstdint>
 #include <string>
 
 namespace NEO {
