From a7fa651bfc431b6df87177bfa38b55147a071122 Mon Sep 17 00:00:00 2001
From: Peter Jung <admin@ptr1337.dev>
Date: Sun, 14 Jul 2024 20:21:14 +0200
Subject: [PATCH] install_user_scheds: Skip packaging of scx_mitosis

Signed-off-by: Peter Jung <admin@ptr1337.dev>
---
 meson-scripts/install_rust_user_scheds | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/meson-scripts/install_rust_user_scheds b/meson-scripts/install_rust_user_scheds
index b3e738c7..fb527245 100755
--- a/meson-scripts/install_rust_user_scheds
+++ b/meson-scripts/install_rust_user_scheds
@@ -7,6 +7,12 @@ for manifest in "$MESON_SOURCE_ROOT"/scheds/rust/*/Cargo.toml; do
     target_dir="${MESON_BUILD_ROOT}${source_dir#${MESON_SOURCE_ROOT}}"
     name="${target_dir##*/}"
 
+    # Skip scx_mitosis
+    if [ "$name" = "scx_mitosis" ]; then
+        echo "Skipping installation of $name"
+        continue
+    fi
+
     bins=($(ls -t "${target_dir}/"*"/${name}"))
     if [ ${#bins[@]} -lt 1 ]; then
 	echo "Cannot find a binary for $name under $target_dir" 1>&2
