From 0f4f8ae377fc6e5f5e92f0e732919a94ea6e6b07 Mon Sep 17 00:00:00 2001
From: Ivan Shapovalov <intelfx@intelfx.name>
Date: Sun, 23 Oct 2016 10:21:54 +0300
Subject: [PATCH 2/9] Makefile: general fixes

---
 Makefile | 58 ++++++++++++++++++++++++++++----------------------------
 1 file changed, 29 insertions(+), 29 deletions(-)

diff --git a/Makefile b/Makefile
index 2906fec..f18459f 100644
--- a/Makefile
+++ b/Makefile
@@ -43,7 +43,7 @@ FOODB=$(PREFIX)/share/foomatic/db/source
 
 # User ID's
 LPuid=-oroot
-LPgid=-glp
+LPgid=-groot
 ifeq ($(UNAME),Darwin)
     LPuid=-oroot
     LPgid=-gwheel
@@ -395,8 +395,8 @@ JBGOPTS=-m 16 -d 0 -p 92	# Equivalent options for pbmtojbg
 #
 # The usual build rules
 #
-all:	all-test $(PROGS) $(BINPROGS) $(SHELLS) getweb \
-	all-icc2ps all-osx-hotplug man doc \
+all:	$(PROGS) $(BINPROGS) $(SHELLS) getweb \
+	all-icc2ps man doc \
 	all-done
 
 all-test:
@@ -629,8 +629,8 @@ command2foo2lava-pjl.o: command2foo2lava-pjl.c
 #
 # Installation rules
 #
-install: all install-test install-prog install-icc2ps install-osx-hotplug \
-	    install-extra install-crd install-foo install-ppd \
+install: all install-prog install-icc2ps install-osx-hotplug \
+	    install-extra install-crd install-ppd \
 	    install-gui install-desktop install-filter \
 	    install-man install-doc
 	#
@@ -679,8 +679,8 @@ install-prog:
 	$(INSTALL) -d $(BIN)
 	$(INSTALL) -c $(PROGS) $(SHELLS) $(BIN)/
 	if [ "$(BINPROGS)" != "" ]; then \
-	    $(INSTALL) -d $(UDEVBIN); \
-	    $(INSTALL) -c $(BINPROGS) $(DESTDIR)/usr/bin/; \
+	    $(INSTALL) -d $(BIN); \
+	    $(INSTALL) -c $(BINPROGS) $(BIN); \
 	fi
 	#
 	# Install gamma correction files.  These are just templates,
@@ -760,7 +760,7 @@ install-crd:
 	# Install prebuilt CRD files (from m2300w project)
 	#
 	$(INSTALL) -d $(SHAREZJS)/
-	$(INSTALL) $(LPuid) $(LPgid) -m 775 -d $(SHAREZJS)/crd/
+	$(INSTALL) $(LPuid) $(LPgid) -m 755 -d $(SHAREZJS)/crd/
 	for i in crd/zjs/*.*; do \
 	    $(INSTALL) -c -m 644 $$i $(SHAREZJS)/crd/; \
 	done
@@ -768,7 +768,7 @@ install-crd:
 	# Install prebuilt CRD files for CLP-300/CLP-600
 	#
 	$(INSTALL) -d $(SHAREQPDL)/
-	$(INSTALL) $(LPuid) $(LPgid) -m 775 -d $(SHAREQPDL)/crd/
+	$(INSTALL) $(LPuid) $(LPgid) -m 755 -d $(SHAREQPDL)/crd/
 	for i in crd/qpdl/*cms* crd/qpdl/*.ps; do \
 	    $(INSTALL) -c -m 644 $$i $(SHAREQPDL)/crd/; \
 	done
@@ -778,7 +778,7 @@ install-psfiles:
 	# Install prebuilt psfiles files (from m2300w project)
 	#
 	$(INSTALL) -d $(SHAREHP)/
-	$(INSTALL) $(LPuid) $(LPgid) -m 775 -d $(SHAREHP)/psfiles/
+	$(INSTALL) $(LPuid) $(LPgid) -m 755 -d $(SHAREHP)/psfiles/
 	for i in psfiles/*.*; do \
 	    $(INSTALL) -c -m 644 $$i $(SHAREHP)/psfiles/; \
 	done
@@ -792,14 +792,14 @@ install-extra:
 	#
 	$(INSTALL) -d $(SHAREZJS)/
 	# foo2zjs ICM files (if any)
-	$(INSTALL) $(LPuid) $(LPgid) -m 775 -d $(SHAREZJS)/icm/
+	$(INSTALL) $(LPuid) $(LPgid) -m 755 -d $(SHAREZJS)/icm/
 	for i in DL*.icm CP*.icm km2430*.icm hp-cp1025*.icm; do \
 	    if [ -f $$i ]; then \
 		$(INSTALL) -c -m 644 $$i $(SHAREZJS)/icm/; \
 	    fi; \
 	done
 	# foo2zjs Firmware files (if any)
-	$(INSTALL) $(LPuid) $(LPgid) -m 775 -d $(SHAREZJS)/firmware/
+	$(INSTALL) $(LPuid) $(LPgid) -m 755 -d $(SHAREZJS)/firmware/
 	for i in sihp1*.img; do \
 	    if [ -f $$i ]; then \
 		base=`basename $$i .img`; \
@@ -808,7 +808,7 @@ install-extra:
 	    fi; \
 	done
 	# foo2xqx Firmware files (if any)
-	$(INSTALL) $(LPuid) $(LPgid) -m 775 -d $(SHAREXQX)/firmware/
+	$(INSTALL) $(LPuid) $(LPgid) -m 755 -d $(SHAREXQX)/firmware/
 	for i in sihpP*.img; do \
 	    if [ -f $$i ]; then \
 		base=`basename $$i .img`; \
@@ -817,49 +817,49 @@ install-extra:
 	    fi; \
 	done
 	# foo2oak ICM files (if any)
-	$(INSTALL) $(LPuid) $(LPgid) -m 775 -d $(SHAREOAK)/icm/
+	$(INSTALL) $(LPuid) $(LPgid) -m 755 -d $(SHAREOAK)/icm/
 	for i in hpclj2[56]*.icm; do \
 	    if [ -f $$i ]; then \
 		$(INSTALL) -c -m 644 $$i $(SHAREOAK)/icm/; \
 	    fi; \
 	done
 	# foo2hp ICM files (if any)
-	$(INSTALL) $(LPuid) $(LPgid) -m 775 -d $(SHAREHP)/icm/
+	$(INSTALL) $(LPuid) $(LPgid) -m 755 -d $(SHAREHP)/icm/
 	for i in hpclj26*.icm km2430*.icm hp1215*.icm; do \
 	    if [ -f $$i ]; then \
 		$(INSTALL) -c -m 644 $$i $(SHAREHP)/icm/; \
 	    fi; \
 	done
 	# foo2lava ICM files (if any)
-	$(INSTALL) $(LPuid) $(LPgid) -m 775 -d $(SHARELAVA)/icm/
+	$(INSTALL) $(LPuid) $(LPgid) -m 755 -d $(SHARELAVA)/icm/
 	for i in km-1600*.icm km2530*.icm; do \
 	    if [ -f $$i ]; then \
 		$(INSTALL) -c -m 644 $$i $(SHARELAVA)/icm/; \
 	    fi; \
 	done
 	# foo2qpdl ICM files (if any)
-	$(INSTALL) $(LPuid) $(LPgid) -m 775 -d $(SHAREQPDL)/icm/
+	$(INSTALL) $(LPuid) $(LPgid) -m 755 -d $(SHAREQPDL)/icm/
 	for i in samclp*.icm; do \
 	    if [ -f $$i ]; then \
 		$(INSTALL) -c -m 644 $$i $(SHAREQPDL)/icm/; \
 	    fi; \
 	done
 	# foo2slx ICM files (if any)
-	$(INSTALL) $(LPuid) $(LPgid) -m 775 -d $(SHARESLX)/icm/
+	$(INSTALL) $(LPuid) $(LPgid) -m 755 -d $(SHARESLX)/icm/
 	for i in lex*.icm; do \
 	    if [ -f $$i ]; then \
 		$(INSTALL) -c -m 644 $$i $(SHARESLX)/icm/; \
 	    fi; \
 	done
 	# foo2hiperc ICM files (if any)
-	$(INSTALL) $(LPuid) $(LPgid) -m 775 -d $(SHAREHC)/icm/
+	$(INSTALL) $(LPuid) $(LPgid) -m 755 -d $(SHAREHC)/icm/
 	for i in OK*.icm C3400*.icm; do \
 	    if [ -f $$i ]; then \
 		$(INSTALL) -c -m 644 $$i $(SHAREHC)/icm/; \
 	    fi; \
 	done
 	# foo2hbpl ICM files (if any)
-	$(INSTALL) $(LPuid) $(LPgid) -m 775 -d $(SHAREHBPL)/icm/
+	$(INSTALL) $(LPuid) $(LPgid) -m 755 -d $(SHAREHBPL)/icm/
 	for i in hbpl*.icm; do \
 	    if [ -f $$i ]; then \
 		$(INSTALL) -c -m 644 $$i $(SHAREHBPL)/icm/; \
@@ -883,14 +883,14 @@ install-ppd:
 	# Install PPD files for CUPS
 	#
 	export PATH=$$PATH:`pwd`:; \
-	if [ -x /usr/sbin/ppdmgr -a -s $(VARPPD)/ppdcache ]; then \
+	if [ -x /usr/bin/ppdmgr -a -s $(VARPPD)/ppdcache ]; then \
 	    $(INSTALL) $(LPgid) -d $(VARPPD)/user; \
 	    cd PPD; \
 	    for ppd in *.ppd; do \
 		manuf=`echo "$$ppd" | sed 's/-.*//'`; \
 		$(INSTALL) $(LPgid) -d $(VARPPD)/user/$$manuf; \
 		modify-ppd <$$ppd | gzip > $(VARPPD)/user/$$manuf/$$ppd.gz; \
-		chmod 664 $(VARPPD)/user/$$manuf/$$ppd.gz; \
+		chmod 644 $(VARPPD)/user/$$manuf/$$ppd.gz; \
 	    done; \
 	    ppdmgr -u; \
 	elif [ -d $(PPD) ]; then \
@@ -907,7 +907,7 @@ install-ppd:
 	    cd PPD; \
 	    for ppd in *.ppd; do \
 		modify-ppd <$$ppd | gzip > $(PPD)/foo2zjs/$$ppd.gz; \
-		chmod 664 $(PPD)/foo2zjs/$$ppd.gz; \
+		chmod 644 $(PPD)/foo2zjs/$$ppd.gz; \
 	    done; \
 	fi
 	#
@@ -917,21 +917,21 @@ install-ppd:
 	    cd PPD; \
 	    for ppd in *.ppd; do \
 		modify-ppd <$$ppd | gzip > $(MODEL)/$$ppd.gz; \
-		chmod 664 $(MODEL)/$$ppd.gz; \
+		chmod 644 $(MODEL)/$$ppd.gz; \
 	    done; \
 	elif [ -d $(LOCALMODEL) ]; then \
 	    rm -f $(LOCALMODEL)/KonicaMinolta*; \
 	    cd PPD; \
 	    for ppd in *.ppd; do \
 		modify-ppd <$$ppd | gzip > $(LOCALMODEL)/$$ppd.gz; \
-		chmod 664 $(LOCALMODEL)/$$ppd.gz; \
+		chmod 644 $(LOCALMODEL)/$$ppd.gz; \
 	    done; \
 	elif [ -d $(MACMODEL) ]; then \
 	    rm -f $(MACMODEL)/KonicaMinolta*; \
 	    cd PPD; \
 	    for ppd in *.ppd; do \
 		modify-ppd <$$ppd | gzip > $(MACMODEL)/$$ppd.gz; \
-		chmod 664 $(MACMODEL)/$$ppd.gz; \
+		chmod 644 $(MACMODEL)/$$ppd.gz; \
 	    done; \
 	fi
 
@@ -972,7 +972,7 @@ DEVDDIR=/etc/devd
 ifeq ($(UNAME),Darwin)
 install-hotplug: install-hotplug-test install-hotplug-osx
 else
-install-hotplug: install-hotplug-test install-hotplug-prog
+install-hotplug: install-hotplug-prog
 endif
 
 install-hotplug-test:
@@ -1075,8 +1075,8 @@ install-hotplug-osx:
 
 install-filter:
 	if [ "$(CUPS_SERVERBIN)" != "" ]; then \
-	    $(INSTALL) -d $(CUPS_SERVERBIN)/filter; \
-	    ln -sf $(BIN)/command2foo2lava-pjl $(CUPS_SERVERBIN)/filter/; \
+	    install -d -m 755 $(CUPS_SERVERBIN)/filter; \
+	    ln -sf ../../../bin/command2foo2lava-pjl $(CUPS_SERVERBIN)/filter/; \
 	fi
 
 CUPSDCONF=/etc/cups/cupsd.conf
-- 
2.19.0

