--- a/gradia/backend/ocr.py	2025-11-18 18:25:10.000000000 +0100
+++ b/gradia/backend/ocr.py	2025-12-28 10:09:27.946197422 +0100
@@ -65,7 +65,7 @@ class OCR:
     def __init__(self, window=None):
         self.tesseract_cmd = ocr_tesseract_cmd
         self.original_tessdata_dir = ocr_original_tessdata
-        self.user_tessdata_dir = os.path.expanduser(f"~/.var/app/{app_id}/data/tessdata")
+        self.user_tessdata_dir = os.path.expanduser(f"~/.local/var/app/{app_id}/data/tessdata")
         self.window = window
 
         pytesseract.pytesseract.tesseract_cmd = self.tesseract_cmd
--- a/meson.build	2025-11-18 18:25:10.000000000 +0100
+++ b/meson.build	2025-12-28 10:02:57.020635194 +0100
@@ -62,8 +62,8 @@ else
 endif
 
 # OCR Directories
-OCR_TESSERACT_CMD = '/app/bin/tesseract'
-OCR_ORIGINAL_TESSDATA_DIR = '/app/share/tessdata'
+ocr_tesseract_cmd = get_option('ocr_tesseract_cmd')
+ocr_original_tessdata_dir = get_option('ocr_original_tessdata_dir')
 
 enable_ocr = get_option('enable-ocr')
 
@@ -81,8 +81,8 @@ conf.set('SCHEMAS_DIR', PKGDATA_DIR)
 conf.set('SOURCE_DIR', meson.current_source_dir())
 conf.set('BUILD_DIR', meson.current_build_dir())
 conf.set('ENABLE_OCR', enable_ocr)
-conf.set('OCR_TESSERACT_CMD', OCR_TESSERACT_CMD)
-conf.set('OCR_ORIGINAL_TESSDATA_DIR', OCR_ORIGINAL_TESSDATA_DIR)
+conf.set('OCR_TESSERACT_CMD', ocr_tesseract_cmd)
+conf.set('OCR_ORIGINAL_TESSDATA_DIR', ocr_original_tessdata_dir)
 
 # Install project information
 conf.set('RELEASE_VER', meson.project_version())
--- a/meson.options	2025-11-18 18:25:10.000000000 +0100
+++ b/meson.options	2025-12-28 09:57:03.560744936 +0100
@@ -1 +1,3 @@
 option('enable-ocr', type: 'boolean', value: true, description: 'Enable OCR functionality')
+option('ocr_tesseract_cmd', type: 'string', value: '/app/bin/tesseract', description: 'OCR cmd path')
+option('ocr_original_tessdata_dir', type: 'string', value: '/app/share/tessdata', description: 'OCR data path')
