diff -uprN src/gradia-1.12.1/data/ui/image_stack.blp src.orig/gradia-1.12.1/data/ui/image_stack.blp
--- a/data/ui/image_stack.blp	2026-04-11 13:54:52.512723539 +0200
+++ b/data/ui/image_stack.blp	2026-02-26 14:30:21.000000000 +0100
@@ -94,7 +94,7 @@ template $GradiaImageStack: Adw.Bin {
                       icon-name: "sidebar-show-symbolic";
                       tooltip-text: _("Open Sidebar");
                       action-name: "win.sidebar-shown";
-                      action-target: true;
+                      action-target: "true";
                       visible: false;
 
                       styles [
