diff -rupN SatDump-1.2.2/plugins/inmarsat_support/aero/libacars/list.c SatDump-1.2.2b/plugins/inmarsat_support/aero/libacars/list.c
--- SatDump-1.2.2/plugins/inmarsat_support/aero/libacars/list.c	2024-11-29 07:58:19.000000000 -0500
+++ SatDump-1.2.2b/plugins/inmarsat_support/aero/libacars/list.c	2026-02-17 17:44:46.196037348 -0500
@@ -36,14 +36,14 @@ size_t la_list_length(la_list const *l)
 	return len;
 }
 
-void la_list_foreach(la_list *l, void (*cb)(), void *ctx) {
+void la_list_foreach(la_list *l, void (*cb)(const void* p, void* c), void *ctx) {
 	la_assert(cb != NULL);
 	for(; l != NULL; l = la_list_next(l)) {
 		cb(l->data, ctx);
 	}
 }
 
-void la_list_free_full_with_ctx(la_list *l, void (*node_free)(), void *ctx) {
+void la_list_free_full_with_ctx(la_list *l, void (*node_free)(void* tag, void* c), void *ctx) {
 	if(l == NULL) {
 		return;
 	}
@@ -57,7 +57,7 @@ void la_list_free_full_with_ctx(la_list
 	LA_XFREE(l);
 }
 
-void la_list_free_full(la_list *l, void (*node_free)()) {
+void la_list_free_full(la_list *l, void (*node_free)(void* tag)) {
 	if(l == NULL) {
 		return;
 	}
diff -rupN SatDump-1.2.2/plugins/inmarsat_support/aero/libacars/list.h SatDump-1.2.2b/plugins/inmarsat_support/aero/libacars/list.h
--- SatDump-1.2.2/plugins/inmarsat_support/aero/libacars/list.h	2024-11-29 07:58:19.000000000 -0500
+++ SatDump-1.2.2b/plugins/inmarsat_support/aero/libacars/list.h	2026-02-17 17:44:46.196435018 -0500
@@ -22,10 +22,10 @@ struct la_list {
 la_list *la_list_next(la_list const *l);
 la_list *la_list_append(la_list *l, void *data);
 size_t la_list_length(la_list const *l);
-void la_list_foreach(la_list *l, void (*cb)(), void *ctx);
+void la_list_foreach(la_list *l, void (*cb)(const void* p, void* c), void *ctx);
 void la_list_free(la_list *l);
-void la_list_free_full(la_list *l, void (*node_free)());
-void la_list_free_full_with_ctx(la_list *l, void (*node_free)(), void *ctx);
+void la_list_free_full(la_list *l, void (*node_free)(void* tag));
+void la_list_free_full_with_ctx(la_list *l, void (*node_free)(void* tag, void* c), void *ctx);
 
 #ifdef __cplusplus
 }
diff -rupN SatDump-1.2.2/src-core/common/calibration.h SatDump-1.2.2b/src-core/common/calibration.h
--- SatDump-1.2.2/src-core/common/calibration.h	2024-11-29 07:58:19.000000000 -0500
+++ SatDump-1.2.2b/src-core/common/calibration.h	2026-02-17 17:45:12.531108023 -0500
@@ -3,9 +3,9 @@
 #include <math.h>
 #include <map>
 
-#define c1 1.1910427e-05
-#define c2 1.4387752
-#define e_num 2.7182818
+constexpr double c1 = 1.1910427e-05;
+constexpr double c2 = 1.4387752;
+constexpr double e_num = 2.7182818;
 
 double temperature_to_radiance(double t, double v);
 
diff -rupN SatDump-1.2.2/src-core/libs/sol2/sol.hpp SatDump-1.2.2b/src-core/libs/sol2/sol.hpp
--- SatDump-1.2.2/src-core/libs/sol2/sol.hpp	2024-11-29 07:58:19.000000000 -0500
+++ SatDump-1.2.2b/src-core/libs/sol2/sol.hpp	2026-02-17 17:44:46.199173573 -0500
@@ -6747,13 +6747,13 @@ namespace sol {
 		/// one.
 		///
 		/// \group emplace
-		template <class... Args>
+		/* template <class... Args>
 		T& emplace(Args&&... args) noexcept {
 			static_assert(std::is_constructible<T, Args&&...>::value, "T must be constructible with Args");
 
 			*this = nullopt;
 			this->construct(std::forward<Args>(args)...);
-		}
+		} */
 
 		/// Swaps this optional with the other.
 		///
@@ -28904,4 +28904,4 @@ namespace sol {
 #endif // SOL_HPP
 // end of sol/sol.hpp
 
-#endif // SOL_SINGLE_INCLUDE_HPP
\ No newline at end of file
+#endif // SOL_SINGLE_INCLUDE_HPP
