diff -wbBur dev86-0.16.19.org/dis88/Makefile dev86-0.16.19/dis88/Makefile
--- dev86-0.16.19.org/dis88/Makefile	1998-07-31 17:01:40.000000000 +0400
+++ dev86-0.16.19/dis88/Makefile	2012-10-24 14:46:02.684587469 +0400
@@ -38,7 +38,7 @@
 
 install:	dis88
 	install -m 755 -s dis88   $(DIST)$(PREFIX)/bin/dis86
-	install -m 644    dis88.1 $(DIST)$(PREFIX)/man/man1/dis86.1
+	install -m 644    dis88.1 $(DIST)$(PREFIX)/share/man/man1/dis86.1
 
 $(OBJ): dis.h a.out.h
 
diff -wbBur dev86-0.16.19.org/Makefile dev86-0.16.19/Makefile
--- dev86-0.16.19.org/Makefile	2012-08-14 00:10:08.000000000 +0400
+++ dev86-0.16.19/Makefile	2012-10-24 14:45:50.917920191 +0400
@@ -17,7 +17,7 @@
 LIBDIR=  $(PREFIX)/lib/bcc
 INCLDIR= $(PREFIX)/lib/bcc
 ASLDDIR= $(BINDIR)
-MANDIR=	 $(PREFIX)/man
+MANDIR=	 $(PREFIX)/share/man
 CFLAGS=  -O
 IFDEFNAME= ifdef
 
diff -wbBur dev86-0.16.19.org/man/Makefile dev86-0.16.19/man/Makefile
--- dev86-0.16.19.org/man/Makefile	1998-07-31 17:03:30.000000000 +0400
+++ dev86-0.16.19/man/Makefile	2012-10-24 14:45:55.494587123 +0400
@@ -1,5 +1,5 @@
 
-MANDIR=/usr/man
+MANDIR=/usr/share/man
 MAN1PG=as86.1 bcc.1 elks.1 elksemu.1 ld86.1
 MAN1DIR=$(MANDIR)/man1
 
