--- a/makeplugin.py
+++ b/makeplugin.py
@@ -34,25 +34,10 @@
                   files,
                   exclude=exclude)
 
-    os.chdir('../included_dependencies')
-    files=[
-        'cloudscraper',
-        'requests',
-        'requests_toolbelt',
-        'requests_file.py',
-        'urllib3',
-        'certifi',
-        'idna',
-        'brotlidecpy', # still needed for cal5.
-        ]
-    createZipFile("../"+filename,"a",
-                  files,
-                  exclude=exclude)
-
     os.chdir('..')
     # 'a' for append
     files=['fanficfare']
     createZipFile(filename,"a",
                   files,
-                  exclude=exclude)
+                  exclude=exclude + ['*.py'])
 
