diff --git a/src/modules/include/module_common_types_public.h b/src/modules/include/module_common_types_public.h
index 345e45ce..de74bf69 100644
--- a/src/modules/include/module_common_types_public.h
+++ b/src/modules/include/module_common_types_public.h
@@ -10,7 +10,7 @@
 
 #ifndef MODULES_INCLUDE_MODULE_COMMON_TYPES_PUBLIC_H_
 #define MODULES_INCLUDE_MODULE_COMMON_TYPES_PUBLIC_H_
-
+#include <cstdint>
 #include <limits>
 
 #include "absl/types/optional.h"
