diff --git a/Makefile b/Makefile
index 0f85019..54814d9 100755
--- a/Makefile
+++ b/Makefile
@@ -62,42 +62,12 @@ replaced/%.gz: replaced/%
 # this target is to be called from the tarball, not from the git
 # working dir!
 install: all installman
-	mkdir -p $(DESTDIR)$(PREFIX)/bin $(DESTDIR)$(PREFIX)/sbin
-	mkdir -p $(DESTDIR)$(PREFIX)/share/monkeysphere/m $(DESTDIR)$(PREFIX)/share/monkeysphere/mh $(DESTDIR)$(PREFIX)/share/monkeysphere/ma $(DESTDIR)$(PREFIX)/share/monkeysphere/transitions
-	mkdir -p $(DESTDIR)$(ETCPREFIX)/etc/monkeysphere
-	mkdir -p $(DESTDIR)$(PREFIX)/share/doc/monkeysphere
-	printf "Monkeysphere %s\n" $(MONKEYSPHERE_VERSION) > $(DESTDIR)$(PREFIX)/share/monkeysphere/VERSION
-	install replaced/src/monkeysphere $(DESTDIR)$(PREFIX)/bin
-	install replaced/src/monkeysphere-host $(DESTDIR)$(PREFIX)/sbin
-	install replaced/src/monkeysphere-authentication $(DESTDIR)$(PREFIX)/sbin
-	install src/monkeysphere-authentication-keys-for-user $(DESTDIR)$(PREFIX)/share/monkeysphere
-	install -m 0644 src/share/common $(DESTDIR)$(PREFIX)/share/monkeysphere
-	install -m 0644 replaced/src/share/defaultenv $(DESTDIR)$(PREFIX)/share/monkeysphere
-	install -m 0755 src/share/keytrans $(DESTDIR)$(PREFIX)/share/monkeysphere
-	ln -sf ../share/monkeysphere/keytrans $(DESTDIR)$(PREFIX)/bin/pem2openpgp
-	ln -sf ../share/monkeysphere/keytrans $(DESTDIR)$(PREFIX)/bin/openpgp2ssh
-	ln -sf ../share/monkeysphere/keytrans $(DESTDIR)$(PREFIX)/bin/openpgp2pem
-	ln -sf ../share/monkeysphere/keytrans $(DESTDIR)$(PREFIX)/bin/openpgp2spki
+	mkdir -p $(DESTDIR)$(PREFIX)/bin
 	install -m 0755 src/agent-transfer/agent-transfer $(DESTDIR)$(PREFIX)/bin
-	install -m 0744 replaced/src/transitions/* $(DESTDIR)$(PREFIX)/share/monkeysphere/transitions
-	install -m 0644 src/transitions/README.txt $(DESTDIR)$(PREFIX)/share/monkeysphere/transitions
-	install -m 0644 src/share/m/* $(DESTDIR)$(PREFIX)/share/monkeysphere/m
-	install -m 0644 src/share/mh/* $(DESTDIR)$(PREFIX)/share/monkeysphere/mh
-	install -m 0644 src/share/ma/* $(DESTDIR)$(PREFIX)/share/monkeysphere/ma
-	install -m 0644 Changelog $(DESTDIR)$(PREFIX)/share/doc/monkeysphere
-	install -d $(DESTDIR)$(PREFIX)/share/doc/monkeysphere/examples
-	install -m 0644 examples/* $(DESTDIR)$(PREFIX)/share/doc/monkeysphere/examples
-	install -m 0644 etc/monkeysphere.conf $(DESTDIR)$(ETCPREFIX)/etc/monkeysphere/monkeysphere.conf$(ETCSUFFIX)
-	install -m 0644 etc/monkeysphere-host.conf $(DESTDIR)$(ETCPREFIX)/etc/monkeysphere/monkeysphere-host.conf$(ETCSUFFIX)
-	install -m 0644 etc/monkeysphere-authentication.conf $(DESTDIR)$(ETCPREFIX)/etc/monkeysphere/monkeysphere-authentication.conf$(ETCSUFFIX)
 
 installman: $(REPLACED_COMPRESSED_MANPAGES)
-	mkdir -p $(DESTDIR)$(MANPREFIX)/man1 $(DESTDIR)$(MANPREFIX)/man7 $(DESTDIR)$(MANPREFIX)/man8
-	install replaced/man/man1/* $(DESTDIR)$(MANPREFIX)/man1
-	install replaced/man/man7/* $(DESTDIR)$(MANPREFIX)/man7
-	install replaced/man/man8/* $(DESTDIR)$(MANPREFIX)/man8
-	ln -sf openpgp2ssh.1.gz $(DESTDIR)$(MANPREFIX)/man1/openpgp2pem.1.gz
-	ln -sf openpgp2ssh.1.gz $(DESTDIR)$(MANPREFIX)/man1/openpgp2spki.1.gz
+	mkdir -p $(DESTDIR)$(MANPREFIX)/man1
+	install replaced/man/man1/agent-transfer.1.gz $(DESTDIR)$(MANPREFIX)/man1
 
 # this target depends on you having the monkeysphere-docs
 # repo checked out as a peer of your monkeysphere repo.
