diff --unified --recursive --text a/Cargo.toml b/Cargo.toml
--- a/Cargo.toml	2026-01-20 13:38:51.731933630 -0700
+++ b/Cargo.toml	2026-01-20 13:41:26.529798747 -0700
@@ -52,3 +52,6 @@
 # libcosmic = { path = "../../cosmic/libcosmic" }
 # cosmic-config = { path = "../../cosmic/libcosmic/cosmic-config" }
 # cosmic-theme = { path = "../../cosmic/libcosmic/cosmic-theme" }
+
+[profile.release]
+lto = "thin"
