diff --unified --recursive --text a/Cargo.toml b/Cargo.toml
--- a/Cargo.toml	2024-12-09 15:47:36.952203029 -0700
+++ b/Cargo.toml	2024-12-09 15:55:12.345871472 -0700
@@ -57,3 +57,6 @@
 # [patch."https://github.com/pop-os/launcher/"]
 # pop-launcher = { git = "https://github.com/wiiznokes/launcher/", rev = "86a54d54a68b832d404ef83a03af84cf9ef3e694" }
 # pop-launcher-service = { git = "https://github.com/wiiznokes/launcher/", rev = "86a54d54a68b832d404ef83a03af84cf9ef3e694" }
+
+[profile.release]
+lto = "thin"
