--- a/libavformat/avisynth.c
+++ b/libavformat/avisynth.c
@@ -120,7 +120,7 @@ typedef struct AviSynthContext {
 
 static av_cold int avisynth_load_library(AviSynthContext *avs)
 {
-    avs->avs_library.library = dlopen(AVISYNTH_LIB, RTLD_NOW | RTLD_LOCAL);
+    avs->avs_library.library = dlopen(AVISYNTH_LIB, RTLD_NOW | RTLD_DEEPBIND);
     if (!avs->avs_library.library)
         return AVERROR_UNKNOWN;
 
