diff --git a/Makefile b/Makefile
index 4359866..854a473 100644
--- a/Makefile
+++ b/Makefile
@@ -8,13 +8,12 @@ CFLAGS+=-O2 -Wall -g
 all: make-bcache probe-bcache bcache-super-show bcache-register bcache
 
 install: make-bcache probe-bcache bcache-super-show
-	$(INSTALL) -m0755 make-bcache bcache-super-show	bcache $(DESTDIR)${PREFIX}/sbin/
-	$(INSTALL) -m0755 probe-bcache bcache-register		$(DESTDIR)$(UDEVLIBDIR)/
-	$(INSTALL) -m0644 69-bcache.rules	$(DESTDIR)$(UDEVLIBDIR)/rules.d/
+	$(INSTALL) -m0755 make-bcache bcache-super-show $(DESTDIR)${PREFIX}/bin/
+	$(INSTALL) -m0755 probe-bcache bcache-register          $(DESTDIR)${PREFIX}$(UDEVLIBDIR)/
+	$(INSTALL) -m0644 69-bcache.rules       $(DESTDIR)${PREFIX}$(UDEVLIBDIR)/rules.d/
 	$(INSTALL) -m0644 -- *.8 $(DESTDIR)${PREFIX}/share/man/man8/
-	$(INSTALL) -D -m0755 initramfs/hook	$(DESTDIR)/usr/share/initramfs-tools/hooks/bcache
 	$(INSTALL) -D -m0755 initcpio/install	$(DESTDIR)/usr/lib/initcpio/install/bcache
-	$(INSTALL) -D -m0755 dracut/module-setup.sh $(DESTDIR)$(DRACUTLIBDIR)/modules.d/90bcache/module-setup.sh
+	$(INSTALL) -D -m0755 dracut/module-setup.sh $(DESTDIR)${PREFIX}$(DRACUTLIBDIR)/modules.d/90bcache/module-setup.sh
 #	$(INSTALL) -m0755 bcache-test $(DESTDIR)${PREFIX}/sbin/
 
 clean:
