diff -rauN acidrip-0.14.orig/AcidRip/acidrip.pm acidrip-0.14/AcidRip/acidrip.pm
--- acidrip-0.14.orig/AcidRip/acidrip.pm	2004-07-25 16:03:09.000000000 +0200
+++ acidrip-0.14/AcidRip/acidrip.pm	2005-10-28 18:46:48.429939022 +0200
@@ -218,7 +218,9 @@
     $menc{'video'} .= ":pass=$::settings->{'video_pass'}" if $::settings->{'video_passes'} > 1;
   }
   if ( $::settings->{'video_codec'} eq 'xvid' ) {
-    $menc{'video'} = "-ovc xvid -xvidencopts $::settings->{'xvid_options'}:bitrate=$::settings->{'video_bitrate'}";
+    $menc{'video'} = "-ovc xvid -xvidencopts ";
+    $menc{'video'} .= "$::settings->{'xvid_options'}:" if $::settings->{'xvid_options'} ne '';
+    $menc{'video'} .= "bitrate=$::settings->{'video_bitrate'}";
     $menc{'video'} .= ":pass=$::settings->{'video_pass'}" if $::settings->{'video_passes'} > 1;
   }
   if ( $::settings->{'video_codec'} eq 'nuv' ) {
