diff --git a/drivers/dahdi/Kbuild b/drivers/dahdi/Kbuild
--- a/drivers/dahdi/Kbuild
+++ b/drivers/dahdi/Kbuild
@@ -144,7 +144,6 @@
     $(shell touch $(KBUILD_EXTMOD)/vpmadt032_loader/.vpmadt032_$(DAHDI_ARCH).o.cmd)
     VPMADT032_LOADER_PRESENT=yes
     dahdi_vpmadt032_loader-objs += vpmadt032_loader/vpmadt032_$(DAHDI_ARCH).o
-    obj-$(DAHDI_BUILD_ALL)$(CONFIG_DAHDI_VPMADT032_LOADER)	+= dahdi_vpmadt032_loader.o
   endif
 endif
