diff --unified --recursive --text os-prober-1.81.orig/linux-boot-probes/mounted/common/90fallback os-prober-1.81/linux-boot-probes/mounted/common/90fallback
--- os-prober-1.81.orig/linux-boot-probes/mounted/common/90fallback	2022-05-26 15:21:10.000000000 -0600
+++ os-prober-1.81/linux-boot-probes/mounted/common/90fallback	2022-07-18 10:22:13.588963249 -0600
@@ -33,7 +33,7 @@
 			# Dracut initramfses are named differently again.
 			initrdname3=$(echo "$kernfile" | sed "s/vmlinu[zx]/initramfs\*/" | sed 's/$/.img/')
 			# And Gentoo's also
-			initrdname4=$(echo "$kernfile" | sed "s/kernel/initramfs\*/")
+			initrdname4=$(echo "$kernfile" | sed "s/kernel\|vmlinu[zx]/initramfs\*/")
 			# Also Alpine
 			initrdname5=$(echo "$kernfile" | sed "s/vmlinu[zx]/initramfs\*/")
 			foundinitrd=0
