From 55eeeec7b4d2fb4db6020bc2dfcd62a40aa16586 Mon Sep 17 00:00:00 2001
From: Tim Van Rillaer <tim.van.rillaer@hotmail.com>
Date: Tue, 20 Dec 2022 18:22:59 +0100
Subject: [PATCH 1/2] Change np.float to float

---
 museval/aggregate.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/museval/aggregate.py b/museval/aggregate.py
index d414ced..bea39a3 100644
--- a/museval/aggregate.py
+++ b/museval/aggregate.py
@@ -126,7 +126,7 @@ def __repr__(self):
             for metric in ['SDR', 'SIR', 'ISR', 'SAR']:
                 out += metric + ":" + \
                     "{:>8.3f}".format(
-                        self.frames_agg([np.float(f['metrics'][metric])
+                        self.frames_agg([float(f['metrics'][metric])
                                          for f in t['frames']])
                     ) + "  "
             out += "\n"

From 48f0db45f1c21a95e7e87adb7562225c0ee89bc1 Mon Sep 17 00:00:00 2001
From: Tim Van Rillaer <tim.van.rillaer@hotmail.com>
Date: Tue, 20 Dec 2022 19:01:07 +0100
Subject: [PATCH 2/2] Change np.float to float

---
 museval/metrics.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/museval/metrics.py b/museval/metrics.py
index 7bfb78b..f52a23b 100644
--- a/museval/metrics.py
+++ b/museval/metrics.py
@@ -551,7 +551,7 @@ def _compute_projection_filters(G, sf, estimated_source):
     filters_len-1
     """
     # epsilon
-    eps = np.finfo(np.float).eps
+    eps = np.finfo(float).eps
 
     # shapes
     (nsampl, nchan) = estimated_source.shape
