diff --git a/CMakeLists.txt b/CMakeLists.txt
index 09c11a9..1211935 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -54,8 +54,7 @@ include_directories(
   ${PROJECT_SOURCE_DIR}/generated/
   ${PROJECT_SOURCE_DIR}/third_party/
   ${PROJECT_SOURCE_DIR}/third_party/Catch2/single_include
-  ${PROJECT_SOURCE_DIR}/third_party/yaml-cpp/include
-  ${PROJECT_SOURCE_DIR}/third_party/antlr4/runtime/Cpp/runtime/src
+  /usr/include/antlr4-runtime
   ${PROJECT_SOURCE_DIR}/src/
   )
 
@@ -77,8 +76,6 @@ file(GLOB_RECURSE yaml-cpp-src
   ${PROJECT_SOURCE_DIR}/third_party/yaml-cpp/src/*.cpp
   )
 
-add_library (antlr4-cpp-runtime ${antlr4-cpp-src})
-add_library (yaml-cpp ${yaml-cpp-src})
 
 add_executable(lua-format ${src_dir} src/main.cpp)
 
@@ -89,7 +86,7 @@ endif()
 
 set_target_properties(lua-format PROPERTIES LINKER_LANGUAGE CXX)
 
-target_link_libraries(lua-format yaml-cpp antlr4-cpp-runtime ${extra-libs})
+target_link_libraries(lua-format yaml-cpp antlr4-runtime ${extra-libs})
 
 install(TARGETS lua-format
   RUNTIME DESTINATION bin
