diff --git a/CMakeLists.txt b/CMakeLists.txt
index d78b82e2c..d1067e251 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -60,6 +60,7 @@ option(USE_DX11 "Build with Direct3D 11 support" ON)
 option(LIBRETRO "Build libretro core" OFF)
 option(USE_OPENGL "Use OpenGL API" ON)
 option(USE_VIDEOCORE "RPI: use the legacy Broadcom GLES libraries" OFF)
+option(USE_BREAKPAD "Build breakpad client and dump symbols" OFF)
 option(APPLE_BREAKPAD "macOS: Build breakpad client and dump symbols" OFF)
 option(ENABLE_GDB_SERVER "Build with GDB debugging support" OFF)
 option(ENABLE_DC_PROFILER "Build with support for target machine (SH4) profiler" OFF)
@@ -247,7 +248,7 @@ if(NINTENDO_SWITCH)
 	target_include_directories( PRIVATE shell/switch "/portlibs/switch/include")
 endif()
 
-if(NOT LIBRETRO)
+if((NOT LIBRETRO) AND (USE_BREAKPAD))
 	if((WIN32 AND NOT MSVC) OR (UNIX AND NOT APPLE AND NOT CMAKE_SYSTEM_NAME MATCHES "(FreeBSD|OpenBSD|NetBSD)"))
 		add_subdirectory(core/deps/breakpad)
 		if(WIN32)
