diff --git a/tests/dependencies/test_closure.py b/tests/dependencies/test_closure.py
index 6231438cd..b5f2b0515 100644
--- a/tests/dependencies/test_closure.py
+++ b/tests/dependencies/test_closure.py
@@ -107,7 +107,7 @@ class Package:
                 for filename in os.listdir(sitepackages):
                     if fnmatch.fnmatch(
                         filename, f"{snake_case_name}-*.dist-info"
-                    ):
+                    ) or fnmatch.fnmatch(filename, f"{snake_case_name}-*.egg-info"):
                         return importlib.metadata.Distribution.at(
                             os.path.join(sitepackages, filename)
                         )
