--- a/Telegram/SourceFiles/payments/ui/payments_reaction_box.cpp
+++ b/Telegram/SourceFiles/payments/ui/payments_reaction_box.cpp
@@ -541,20 +541,29 @@
 		[=] { box->closeBox(); });
 
 	const auto addTopReactors = [&] {
-		FillTopReactors(
-			content,
-			std::move(args.top),
-			colorings,
-			(videoStreamAdmin
-				? rpl::single(state->chosen.current())
-				: state->chosen.value()),
-			(videoStreamAdmin
-				? rpl::single(state->shownPeer.current())
-				: state->shownPeer.value()),
-			[=](uint64 barePeerId) {
-				state->shownPeer = state->savedShownPeer = barePeerId;
-			},
-			videoStream);
+		if (videoStreamAdmin) {
+			FillTopReactors(
+				content,
+				std::move(args.top),
+				colorings,
+				rpl::single(state->chosen.current()),
+				rpl::single(state->shownPeer.current()),
+				[=](uint64_t barePeerId) {
+					state->shownPeer = state->savedShownPeer = barePeerId;
+				},
+				videoStream);
+		} else {
+			FillTopReactors(
+				content,
+				std::move(args.top),
+				colorings,
+				state->chosen.value(),
+				state->shownPeer.value(),
+				[=](uint64_t barePeerId) {
+					state->shownPeer = state->savedShownPeer = barePeerId;
+				},
+				videoStream);
+		}
 	};
 
 	if (videoStreamChoosing) {
