diff --unified --recursive --text a/src/Dialogs/Preferences/PreferencesWindow.vala b/src/Dialogs/Preferences/PreferencesWindow.vala
--- a/src/Dialogs/Preferences/PreferencesWindow.vala	2025-08-09 09:52:35.696226452 -0600
+++ b/src/Dialogs/Preferences/PreferencesWindow.vala	2025-08-09 09:54:11.994475682 -0600
@@ -888,11 +888,7 @@
     private Adw.NavigationPage get_quick_add_page () {
         var settings_header = new Dialogs.Preferences.SettingsHeader (_("Quick Add"));
 
-        string quick_add_command =
-            "flatpak run --command=io.github.alainm23.planify.quick-add %s".printf (Build.APPLICATION_ID);
-        if (GLib.Environment.get_variable ("SNAP") != null) {
-            quick_add_command = "planify.quick-add";
-        }
+        string quick_add_command = "planify.quick-add";
 
         var description_label = new Gtk.Label (
             _(
