From 6b948668b9a700aa86ef7c5843d8ab10b668ebf6 Mon Sep 17 00:00:00 2001
From: Vendicated <vendicated@riseup.net>
Date: Thu, 23 May 2024 17:40:57 +0200
Subject: [PATCH] make package manager check warn instead of error on
 mismatched version

---
 .npmrc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/.npmrc b/.npmrc
index d67f3748..eee7e86f 100644
--- a/.npmrc
+++ b/.npmrc
@@ -1 +1,2 @@
 node-linker=hoisted
+package-manager-strict=false
\ No newline at end of file
