diff --unified --recursive --text a/Cargo.toml b/Cargo.toml
--- a/Cargo.toml	2026-03-24 09:20:38.341371213 -0600
+++ b/Cargo.toml	2026-03-24 09:21:33.531670491 -0600
@@ -55,3 +55,6 @@
 # iced_futures = { git = "https://github.com/pop-os/libcosmic//" }
 # cosmic-config = { git = "https://github.com/pop-os/libcosmic//" }
 # cosmic-theme = { git = "https://github.com/pop-os/libcosmic//" }
+
+[profile.release]
+lto = "thin"
