diff --git i/lib/hyprland/meson.build w/lib/hyprland/meson.build
index ff53601..7efda2f 100644
--- i/lib/hyprland/meson.build
+++ w/lib/hyprland/meson.build
@@ -57,9 +57,10 @@ if get_option('lib')
     dependencies: deps,
     vala_header: meson.project_name() + '.h',
     vala_vapi: meson.project_name() + '-' + api_version + '.vapi',
+    vala_gir: gir,
     version: meson.project_version(),
     install: true,
-    install_dir: [true, true, true],
+    install_dir: [true, true, true, true],
   )
 
   import('pkgconfig').generate(
@@ -77,17 +78,17 @@ if get_option('lib')
     pkgs += ['--pkg=' + dep.name()]
   endforeach
 
-  gir_tgt = custom_target(
-    gir,
-    command: [find_program('python3'), files('gir.py'), meson.project_name(), gir]
-    + pkgs
-    + sources,
-    input: sources,
-    depends: lib,
-    output: gir,
-    install: true,
-    install_dir: get_option('datadir') / 'gir-1.0',
-  )
+  # gir_tgt = custom_target(
+  #   gir,
+  #   command: [find_program('python3'), files('gir.py'), meson.project_name(), gir]
+  #   + pkgs
+  #   + sources,
+  #   input: sources,
+  #   depends: lib,
+  #   output: gir,
+  #   install: true,
+  #   install_dir: get_option('datadir') / 'gir-1.0',
+  # )
   custom_target(
     typelib,
     command: [
@@ -98,7 +99,7 @@ if get_option('lib')
     ],
     input: lib,
     output: typelib,
-    depends: [lib, gir_tgt],
+    depends: [lib],
     install: true,
     install_dir: get_option('libdir') / 'girepository-1.0',
   )
