--- a/meson.build
+++ b/meson.build
@@ -30,8 +30,8 @@
 bash_completion_dep = dependency('bash-completion', required: false)

 cc = meson.get_compiler('c')
-libm_dep = cc.find_library('m')
 valac = meson.get_compiler('vala')
+libm_dep = cc.find_library('m')
 posix_dep = valac.find_library('posix')

 check_headers = [
@@ -107,6 +107,10 @@
   add_project_arguments(['--enable-checking'], language: 'vala')
 endif

+if valac.version().version_compare('>= 0.56.19')
+  add_project_arguments(['-D', 'VALA_0_56_19'], language: 'vala')
+endif
+
 application_id = 'io.github.focustimerhq.FocusTimer'
 application_name = 'Focus Timer'
 package_name = meson.project_name()
--- a/src/ui/preferences/automation/action/action-listboxrow.vala
+++ b/src/ui/preferences/automation/action/action-listboxrow.vala
@@ -91,7 +91,12 @@
             this.drag_widget.append (row);
             this.drag_widget.drag_highlight_row (row);

+            #if VALA_0_56_19
+            var drag_icon = new Gtk.DragIcon.get_for_drag (drag);
+            #else
             var drag_icon = (Gtk.DragIcon) Gtk.DragIcon.get_for_drag (drag);
+            #endif
+
             drag_icon.child = this.drag_widget;

             drag.set_hotspot ((int) this.drag_x, (int) this.drag_y);
