diff --git a/nebula/ui/components/MZFlickable.qml b/nebula/ui/components/MZFlickable.qml
index 0df52bcbca..b0130fa77d 100644
--- a/nebula/ui/components/MZFlickable.qml
+++ b/nebula/ui/components/MZFlickable.qml
@@ -22,15 +22,11 @@ Flickable {
     boundsBehavior: Flickable.StopAtBounds
     opacity: 0
 
-    onFlickContentHeightChanged: {
-        recalculateContentHeight()
-    }
-
-    onHeightChanged: {
-        recalculateContentHeight()
-    }
-
     Component.onCompleted: {
+        recalculateContentHeight()
+        this.onHeightChanged.connect(recalculateContentHeight)
+        this.onFlickContentHeightChanged.connect(recalculateContentHeight)
+    
         opacity = 1;
         if (Qt.platform.os === "windows") {
             maximumFlickVelocity = 700;
diff --git a/src/commands/commandui.cpp b/src/commands/commandui.cpp
index a0042b77f2..7b160c13b6 100644
--- a/src/commands/commandui.cpp
+++ b/src/commands/commandui.cpp
@@ -207,7 +207,7 @@ int CommandUI::run(QStringList& tokens) {
     EventListener eventListener;
 #endif
 
-#ifdef MZ_DEBUG
+#ifdef false
     // This enables the qt-creator qml debugger on debug builds.:
     // Go to QtCreator: Debug->Start Debugging-> Attach to QML port
     // Port is 1234.
diff --git a/src/platforms/windows/daemon/windowsfirewall.cpp b/src/platforms/windows/daemon/windowsfirewall.cpp
index b65e982476..50f1c767a5 100644
--- a/src/platforms/windows/daemon/windowsfirewall.cpp
+++ b/src/platforms/windows/daemon/windowsfirewall.cpp
@@ -22,6 +22,7 @@
 #include <QObject>
 #include <QScopeGuard>
 #include <QtEndian>
+#include <utility>
 
 #include "ipaddress.h"
 #include "leakdetector.h"
@@ -320,7 +321,7 @@ bool WindowsFirewall::disableKillSwitch() {
     FwpmFilterDeleteById0(m_sessionHandle, filterID);
   }
 
-  for (const auto& filterID : qAsConst(m_activeRules)) {
+  for (const auto& filterID : std::as_const(m_activeRules)) {
     FwpmFilterDeleteById0(m_sessionHandle, filterID);
   }
 
