From 2215108e1ac686f6faff0d46441c5adb44e8d7c9 Mon Sep 17 00:00:00 2001
From: "j.r" <j.r@jugendhacker.de>
Date: Thu, 14 Jan 2021 23:57:02 +0100
Subject: [PATCH] Change paths for packaging

---
 youplay.desktop | 2 +-
 youplay.sh      | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/youplay.desktop b/youplay.desktop
index b335943..187dc4c 100755
--- a/youplay.desktop
+++ b/youplay.desktop
@@ -4,7 +4,7 @@ Name=YouPlay
 GenericName=Play music from Youtube
 Exec=youplay --gui=gtk4
 Terminal=false
-Icon=/usr/share/icons/youplay.svg
+Icon=/usr/share/youplay/youplay.svg
 StartupNotify=false
 Categories=GNOME;GTK;Utility;Music
 Keywords=Multimedia;Music;Youtube
diff --git a/youplay.sh b/youplay.sh
index 527850b..31f02df 100755
--- a/youplay.sh
+++ b/youplay.sh
@@ -1,3 +1,3 @@
 #!/bin/bash
-cd `dirname $0`
-python3 youplay.py --gui=gtk4
+cd '/usr/share/youplay/'
+python3 youplay.py --gui=@@gui@@
-- 
2.41.0

