--- a/third_party/CMakeLists.txt
+++ b/third_party/CMakeLists.txt
@@ -14,9 +14,9 @@ if (ENABLE_NPU_PERFETTO_BUILD)
   include(cmake/perfetto.cmake)
 endif()
 
-set(INSTALL_GTEST OFF)
-add_subdirectory(googletest EXCLUDE_FROM_ALL)
+#set(INSTALL_GTEST OFF)
+#add_subdirectory(googletest EXCLUDE_FROM_ALL)
 
-set(YAML_CPP_INSTALL OFF)
-set(BUILD_SHARED_LIBS OFF)
-add_subdirectory(yaml-cpp EXCLUDE_FROM_ALL)
+#set(YAML_CPP_INSTALL OFF)
+#set(BUILD_SHARED_LIBS OFF)
+#add_subdirectory(yaml-cpp EXCLUDE_FROM_ALL)
