diff --git a/cmake/BuildParameters.cmake b/cmake/BuildParameters.cmake
index e16d726cb..97b4a433f 100644
--- a/cmake/BuildParameters.cmake
+++ b/cmake/BuildParameters.cmake
@@ -13,7 +13,7 @@ add_definitions(-DGIT_VERSION="${GIT_VERSION}")
 #-------------------------------------------------------------------------------
 option(LIBRETRO "Enables building the libretro core" ON)
 set(CMAKE_BUILD_PO FALSE)
-set(USE_SYSTEM_LIBS OFF)
+set(USE_SYSTEM_LIBS ON)
 add_definitions(-D__LIBRETRO__)
 optional_system_library(ryml)
 set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
diff --git a/common/StringUtil.h b/common/StringUtil.h
index 5b23d1b9c..d8f0a27f2 100644
--- a/common/StringUtil.h
+++ b/common/StringUtil.h
@@ -30,7 +30,7 @@
 #define HAD_M_ARM64 _M_ARM64
 #undef _M_ARM64
 #endif
-#include "../3rdparty/rapidyaml/rapidyaml/ext/c4core/src/c4/ext/fast_float/include/fast_float/fast_float.h"
+#include <fast_float/fast_float.h>
 #if defined(HAD_M_ARM64) && !defined(_WIN32)
 #define _M_ARM64 HAD_M_ARM64
 #undef HAD_M_ARM64
diff --git a/pcsx2/GameDatabase.cpp b/pcsx2/GameDatabase.cpp
index 24367faa6..4027f36f0 100644
--- a/pcsx2/GameDatabase.cpp
+++ b/pcsx2/GameDatabase.cpp
@@ -18,8 +18,8 @@
 #include <optional>
 
 #include "../common/Threading.h"
-#include "../3rdparty/rapidyaml/rapidyaml/src/ryml_std.hpp"
-#include "../3rdparty/rapidyaml/rapidyaml/src/ryml.hpp"
+#include <ryml_std.hpp>
+#include <ryml.hpp>
 
 #include "../common/Console.h"
 #include "../common/FileSystem.h"
