diff --git a/src/wl_window.c b/src/wl_window.c
index 5b491ffb..05239e02 100644
--- a/src/wl_window.c
+++ b/src/wl_window.c
@@ -2236,16 +2236,16 @@ void _glfwGetWindowPosWayland(_GLFWwindow* window, int* xpos, int* ypos)
     // A Wayland client is not aware of its position, so just warn and leave it
     // as (0, 0)

-    _glfwInputError(GLFW_FEATURE_UNAVAILABLE,
-                    "Wayland: The platform does not provide the window position");
+    fprintf(stderr,
+                    "Wayland: The platform does not provide the window position\n");
 }

 void _glfwSetWindowPosWayland(_GLFWwindow* window, int xpos, int ypos)
 {
     // A Wayland client can not set its position, so just warn

-    _glfwInputError(GLFW_FEATURE_UNAVAILABLE,
-                    "Wayland: The platform does not support setting the window position");
+    fprintf(stderr,
+                    "Wayland: The platform does not support setting the window position\n");
 }

 void _glfwGetWindowSizeWayland(_GLFWwindow* window, int* width, int* height)
