--- wayfire-0.10.1/src/meson.build.orig	2025-12-08 14:59:22.164545413 +0200
+++ wayfire-0.10.1/src/meson.build	2025-12-08 14:59:33.414613844 +0200
@@ -118,7 +118,7 @@
     dependencies: wayfire_dependencies,
     include_directories: [wayfire_conf_inc, wayfire_api_inc],
     cpp_args: debug_arguments,
-    install: false, cpp_pch: 'pch/pch.h')
+    install: false)
 
 libwayfire = declare_dependency(link_whole: libwayfire_sta,
     include_directories: [wayfire_conf_inc, wayfire_api_inc],
