diff --git a/Makefile b/Makefile
index 9e22942..faaa9dd 100644
--- a/Makefile
+++ b/Makefile
@@ -45,7 +45,7 @@ build:
 	$(GODOT) --path . --export "Linux/X11" target/debug/luxtorpeda.x86_64 --no-window
 
 release:
-	cargo build --release
+	cargo build --release --locked
 	touch target/.gdignore
 	mkdir -p target/debug
 	cp -r target/release/* target/debug
@@ -55,7 +55,7 @@ lint:
 	cargo clippy -- -D warnings
 
 test:
-	cargo test
+	cargo test --release --locked
 
 clean:
 	cargo clean
