From be0e6f07a827dd1686d8ae9da2f6d1930f501929 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michael=20La=C3=9F?= <bevan@bi-co.net>
Date: Sat, 4 Nov 2023 12:30:04 +0100
Subject: [PATCH 2/4] Quick & Dirty patch to allow finding ngspice on Arch
 Linux

---
 pri/spicedetect.pri | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/pri/spicedetect.pri b/pri/spicedetect.pri
index 76795bd8..707abdbf 100644
--- a/pri/spicedetect.pri
+++ b/pri/spicedetect.pri
@@ -14,13 +14,12 @@
 # ********************************************************************/
 
 
-message("Using fritzing ngspice detect script.")
-NGSPICEPATH = ../../ngspice-42
-NGSPICEPATH = $$absolute_path($${NGSPICEPATH})
+message("Using customized ngspice detect script.")
+NGSPICEPATH = "/usr/include/ngspice/"
 
 exists($$NGSPICEPATH) {
-	message("found $${NGSPICEPATH}")
-	INCLUDEPATH += $$NGSPICEPATH/include
+	message("using $${NGSPICEPATH}")
+	INCLUDEPATH += $$NGSPICEPATH
 } else {
     error("ngspice not found in $${NGSPICEPATH}")
 }
-- 
2.49.0

