diff --unified --recursive --text a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2024-08-17 09:11:49.440332679 -0600
+++ b/CMakeLists.txt	2024-08-17 09:13:05.617191316 -0600
@@ -495,7 +495,7 @@
         if(${RPM})
             install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/tools/hardinfo2.service DESTINATION /usr/lib/systemd/system/)
         else()
-            install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/tools/hardinfo2.service DESTINATION /lib/systemd/system/)
+            install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/tools/hardinfo2.service DESTINATION ${CMAKE_INSTALL_LIBDIR}/systemd/system/)
         endif()
     endif()
     configure_file(${CMAKE_CURRENT_SOURCE_DIR}/tools/postinst ${CMAKE_CURRENT_BINARY_DIR}/packaging/postinst)
