--- a/ext/enchant/config.m4
+++ b/ext/enchant/config.m4
@@ -14,9 +14,9 @@
 	    ENCHANT_SEARCH_DIRS="/usr/local /usr"
 	fi
 	for i in $ENCHANT_SEARCH_DIRS; do
-		if test -f $i/include/enchant/enchant.h; then
+		if test -f $i/include/enchant-2/enchant.h; then
 			ENCHANT_DIR=$i
-			ENCHANT_INCDIR=$i/include/enchant
+			ENCHANT_INCDIR=$i/include/enchant-2
 		elif test -f $i/include/enchant.h; then
 			ENCHANT_DIR=$i
 			ENCHANT_INCDIR=$i/include
@@ -31,7 +31,7 @@
 
 	AC_DEFINE(HAVE_ENCHANT,1,[ ])
 	PHP_SUBST(ENCHANT_SHARED_LIBADD)
-	PHP_ADD_LIBRARY_WITH_PATH(enchant, $ENCHANT_LIBDIR, ENCHANT_SHARED_LIBADD)
+	PHP_ADD_LIBRARY_WITH_PATH(enchant-2, $ENCHANT_LIBDIR, ENCHANT_SHARED_LIBADD)
 	PHP_ADD_INCLUDE($ENCHANT_INCDIR)
 	PHP_CHECK_LIBRARY(enchant, enchant_broker_set_param,
 	[
