From 8faeee4d2a73b52f5ef16b059e4e666976c7079d Mon Sep 17 00:00:00 2001
From: Stelios Tsampas <loathingkernel@gmail.com>
Date: Sat, 1 Jun 2024 18:59:02 +0300
Subject: [PATCH 2/2] fshack: mark monitor mode structs and fsr ratio floats as
 const

---
 dlls/winex11.drv/fs.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/dlls/winex11.drv/fs.c b/dlls/winex11.drv/fs.c
index b4c93b555c3..d5825e38456 100644
--- a/dlls/winex11.drv/fs.c
+++ b/dlls/winex11.drv/fs.c
@@ -47,7 +47,7 @@ struct fs_monitor_size
     SIZE size;
     BOOL additional;
 };
-static struct fs_monitor_size fs_monitor_sizes_base[] =
+static const struct fs_monitor_size fs_monitor_sizes_base[] =
 {
     {{640, 480}},   /*  4:3 */
     {{800, 600}},   /*  4:3 */
@@ -75,7 +75,7 @@ static struct fs_monitor_size fs_monitor_sizes_base[] =
 };
 
 /* The order should be in sync with the values in 'fs_hack_is_fsr_single_mode'*/
-static float fsr_ratios[] = {
+static const float fsr_ratios[] = {
     2.0f, /* FSR Performance */
     1.7f, /* FSR Balanced */
     1.5f, /* FSR Quality */
-- 
2.45.2

