diff --git a/python/setup.py b/python/setup.py
index 60e0811..a91ccf5 100755
--- a/python/setup.py
+++ b/python/setup.py
@@ -297,7 +297,6 @@ else:
   cmdclass = {'build_ext': build_ext_unix}
 
 if __name__ == '__main__':
-  copy_package_data()
   setup(
       name='sentencepiece',
       package_dir={'': 'src'},
diff --git a/python/src/sentencepiece/__init__.py b/python/src/sentencepiece/__init__.py
index 6126ad2..8576f7c 100644
--- a/python/src/sentencepiece/__init__.py
+++ b/python/src/sentencepiece/__init__.py
@@ -919,8 +919,3 @@ _add_snake_case(SentencePieceNormalizer)
 
 # Global variables and initialization
 from ._version import __version__
-
-try:
-    SetDataDir(os.path.join(str(importlib.resources.files('sentencepiece')), 'package_data'))
-except Exception:
-    pass
