diff --git i/lib/bluetooth/meson.build w/lib/bluetooth/meson.build
index 529e85f..e296ab0 100644
--- i/lib/bluetooth/meson.build
+++ w/lib/bluetooth/meson.build
@@ -50,9 +50,10 @@ lib = library(
   vala_args: ['--vapi-comments'],
   vala_header: meson.project_name() + '.h',
   vala_vapi: meson.project_name() + '-' + api_version + '.vapi',
+  vala_gir: gir,
   version: meson.project_version(),
   install: true,
-  install_dir: [true, true, true],
+  install_dir: [true, true, true, true],
 )
 
 pkgs = []
@@ -60,17 +61,17 @@ foreach dep : deps
   pkgs += ['--pkg=' + dep.name()]
 endforeach
 
-gir_tgt = custom_target(
-  gir,
-  command: [find_program('python3'), files('gir.py'), meson.project_name(), gir]
-  + pkgs
-  + sources,
-  input: sources,
-  depends: lib,
-  output: gir,
-  install: true,
-  install_dir: get_option('datadir') / 'gir-1.0',
-)
+# gir_tgt = custom_target(
+#   gir,
+#   command: [find_program('python3'), files('gir.py'), meson.project_name(), gir]
+#   + pkgs
+#   + sources,
+#   input: sources,
+#   depends: lib,
+#   output: gir,
+#   install: true,
+#   install_dir: get_option('datadir') / 'gir-1.0',
+# )
 
 custom_target(
   typelib,
@@ -82,7 +83,7 @@ custom_target(
   ],
   input: lib,
   output: typelib,
-  depends: [lib, gir_tgt],
+  depends: [lib],
   install: true,
   install_dir: get_option('libdir') / 'girepository-1.0',
 )
