--- a/cmake/developer_package/version.cmake
+++ b/cmake/developer_package/version.cmake
@@ -99,7 +99,7 @@ macro(ov_parse_ci_build_number repo_root)
         endforeach()
 
         foreach(var OpenVINO_VERSION_MAJOR OpenVINO_VERSION_MINOR OpenVINO_VERSION_PATCH)
-            if(DEFINED ${var} AND NOT ${var} EQUAL ${var}_HPP)
+            if(DEFINED ${var} AND NOT ${${var}} EQUAL ${${var}_HPP})
                 message(FATAL_ERROR "${var} parsed from CI_BUILD_NUMBER (${${var}}) \
                     and from openvino/core/version.hpp (${${var}_HPP}) are different")
             else()
