diff --git i/lib/astal/io/meson.build w/lib/astal/io/meson.build
index 9a00904..1133208 100644
--- i/lib/astal/io/meson.build
+++ w/lib/astal/io/meson.build
@@ -51,11 +51,12 @@ lib = library(
   sources,
   dependencies: deps,
   vala_args: ['--vapi-comments'],
+  vala_gir: gir,
   vala_header: meson.project_name() + '.h',
   vala_vapi: meson.project_name() + '-' + api_version + '.vapi',
   version: meson.project_version(),
   install: true,
-  install_dir: [true, true, true],
+  install_dir: [true, true, true, true],
 )
 
 pkgs = []
@@ -63,18 +64,18 @@ foreach dep : deps
   pkgs += ['--pkg=' + dep.name()]
 endforeach
 
-gir_tgt = custom_target(
-  gir,
-  command: [find_program('python3'), files('gir.py'), meson.project_name(), gir]
-  + pkgs
-  + sources,
-  input: sources,
-  depends: lib,
-  output: gir,
-  install: true,
-  install_dir: get_option('datadir') / 'gir-1.0',
-)
-
+# gir_tgt = custom_target(
+#   gir,
+#   command: [find_program('python3'), files('gir.py'), meson.project_name(), gir]
+#   + pkgs
+#   + sources,
+#   input: sources,
+#   depends: lib,
+#   output: gir,
+#   install: true,
+#   install_dir: get_option('datadir') / 'gir-1.0',
+# )
+#
 custom_target(
   typelib,
   command: [
@@ -85,7 +86,7 @@ custom_target(
   ],
   input: lib,
   output: typelib,
-  depends: [lib, gir_tgt],
+  depends: [lib],
   install: true,
   install_dir: get_option('libdir') / 'girepository-1.0',
 )
