diff --git a/plugins/opengl/src/vertexbuffer.cpp b/plugins/opengl/src/vertexbuffer.cpp
index b102016..4e314c2 100644
--- a/plugins/opengl/src/vertexbuffer.cpp
+++ b/plugins/opengl/src/vertexbuffer.cpp
@@ -164,7 +164,7 @@ void GLVertexBuffer::addVertices (GLuint nVertices, const GLfloat *vertices)
 
 GLfloat *GLVertexBuffer::getVertices() const
 {
-    return &priv->vertexData[0];
+    return priv->vertexData.data();
 }
 
 int GLVertexBuffer::getVertexStride() const
